---
author:
  - Luna L. Sanchez Reyes
  - Martha Kandziora
  - Emily Jane McTavish
  - University of California, Merced
csl: apa.csl
output:
  redoc::redoc
keep_tex: true
---

1. Phylogenies are a key part of research in many areas of biology. Tools that automatize
some parts of the process of phylogenetic reconstruction (mainly character matrix construction)
have been developed for the advantage of both specialists in the field of phylogenetics and nonspecialists.
However, interpretation of results, comparison with previously available phylogenetic
hypotheses, and choosing of one phylogeny for downstream analyses and discussion still impose difficulties
to one that is not a specialist either on phylogenetic methods or on a particular group of study.

1. Physcraper is an open‚Äêsource, command-line Python program that automatizes the update of published
phylogenies by enriching underlying gene alignments with public DNA sequence data, and linking taxonomic information across databases.
This provides a framework for comparison of published phylogenies with their updated versions, by using the conflict Application Programming Interface (API) function of the Open Tree of Life project.

1. Physcraper can be used by the nonspecialist, as a tool to generate phylogenetic
hypothesis based on already available expert phylogenetic knowledge.
Phylogeneticists and group specialists will find it useful as a tool to facilitate dataset gathering and comparison
of alternative phylogenetic hypotheses (topologies).

1. We hope the physcraper workflow demonstrates the benefits of opening results in phylogenetics and encourages researchers to strive for better data sharing practices. Physcraper can be used with any OS. Detailed instructions for installation and
use are available at [https://physcraper.readthedocs](https://physcraper.readthedocs.io/en/tutorial/index.html).
