---
title: 'Physcraper: Body'
author:
  - Luna L. Sanchez Reyes
  - Martha Kandziora
  - Emily Jane McTavish
  - University of California, Merced
date: '`r format(Sys.time(), "%d %B, %Y")`'
bibliography: paper.bib
csl: apa.csl
output:
  redoc::redoc
---


# Introduction

From molecular data to alignments and phylogenies, public biological data repositories such as the GenBank database [@benson2000genbank; @wheeler2000database], the TreeBASE repository [@piel2009treebase] and the Open Tree of Life curating system [@mctavish2015phylesystem], are still accumulating data.

More than a decade ago, the National Center for Biodiversity Information (NCBI) molecular database, GenBank, released its version number 159 (April 15, 2007). With 72 million DNA sequences, it was estimated to have the potential to resolve evolutionary relationships of most of the 241 000 distinct taxa represented in it [about `r round(236023/240708, digits=4)*100`% of taxa in the NCBI taxonomy; @sanderson2008phylota], covering about 10% of extant described biodiversity [taking a conservative estimate of extant diversity; @scott2011ncbi; @federhen2003taxonomy].
In comparison, the current GenBank release number 238 (June 15, 2020) has tripled in size, hosting data for more than 217 million DNA sequences <https://ftp.ncbi.nih.gov/genbank/gbrel.txt>.

Many useful tools have been developed in the past decades in an effort to make sense of the large amount of data in public molecular databases, as well as private ones. Generally refered to as "pipelines", most of these tools were motivated by the genomics revolution, to identify clusters of homologs for genomic assembly. Notably, this homolog DNA clusters can be used as homology hypotheses (aka molecular alignments) to reconstruct phylogenetic relationships.
Pipelines that automatize the assembly of DNA alignments from the GenBank database for phylogenetic reconstruction (refered to as "phylogenetic pipelines") such as PHYLOTA [@sanderson2008phylota], PHLAWD [@smith2009mega], and SUPERSMART [@antonelli2017toward], have been widely used to study the evolutionary relationships among different organisms (TABLE?? maybe supplementary data), from a phylogeny of no more than 20 species of the family of barracuda fish [Sphyraenidae; @santini2013first], to a mega-phylogeny of almost 3 000 species of living ferns [@lehtonen2011towards].

Pipelines have been an important incorporation to the field of phylogenetics in many ways, particularly because they represent a clear step towards reproducibility in the field. In contrast, most published phylogenies have been infered using homology hypotheses that have been curated "by hand" [@morrison2009would].
What explains this preference? aka the discordance between the genomics approach (a gazillion markers poorly aligned, but the amount of data will overcome errors in alignment) and the classic phylogenetics approach (few markers thoughtfully curated).
On one hand, there are concerns about the use of automated pipelines in phylogenetics.
Concerns I think people have about these tools:
- Errors in identification of sequences
- Little control at different steps of the process
- Too much of a black box? Most of these phylogenies are being constructed by people learning about the methods (aka students), so they want to know what is going on at each step and end up doing it manually.
-
It has also been suggested that manual curation of classic and genomic alignments produces better phylogenetic reconstructions [@fragoso2017pilot].

A way to incorporate the best of two worlds (massive amounts of newly released molecular data VS fine curation from human experts) would be to rely on published manually curated homology hypotheses. This expert-curated alignments can be enriched and updated by incorporating newly released data from public molecular databases.

As of April 2014, TreeBASE hosted a bit more than 8 200 curated alignments, providing information on evolutionary relationships of almost 105 000 distinct taxa [(see TreeBASE about)](https://www.treebase.org/treebase-web/home.html#:~:text=TreeBASE%20is%20produced%20and%20governed,mapped%20to%20104%2C593%20distinct%20taxa.).
This provides an untapped source of valuable knowledge with the potential to update phylogenetic relationships in different regions of the tree of life.

The OToL tree repository [phylesystem; @mctavish2015phylesystem] automatically incorporates the phylogenies from TreeBASE, and stores metadata linking the tree to its corresponding alignment repository in TreeBASE. This provides a loose means of linking the tree with the exact alignment that generated it. This can be a difficult task when multiple alignments were originally deposited and no clear link between alignment and tree was provided in TreeBASE.

Ultimately, linkage of original alignment with its corresponding phylogeny has to be done by a human curator.
Moreover, each one of these data repositories follow their own system for taxon identification. A real challenge is to find an effective way to link data from across databases that belongs to the same taxon.
OToL's metadata system incorporates taxon identifiers from a variety of taxonomies and repositories, including the NCBI taxonomy, GBIF, etc.

Physcraper is a python pipeline that uses the OToL metadata system to connect databases through taxon identification numbers.

It's main goal is to connect public phylogenetic relationships in OToL, with alignments from TreeBASE and GenBank DNA data.

It also allows automatizing and standardizing the comparison of phylogenetic hypotheses.

This is an effort to keep on directing ourselves towards a fully reproducible workflow in phylogenetics.
And an effort to more effectively connect the world of big data in biology.


# How does Physcraper work?

## The input: a study tree and an alignment

- The study tree is a published phylogenetic tree stored in the OToL database, phylesystem [@mctavish2015phylesystem]. The main
  reason for this is that trees in phylesystem have a set of user defined characteristics
  that are essential for automatizing the phylogeny update process. The most relevant of these being the definition of ingroup and outgroup. Outgroup and ingroup taxa in the original tree are identified and tagged. This allows to automatically set the root for the updated tree on the next steps of the pipeline.
  A user can choose from the `r rotl::tol_about()$num_source_trees` published trees supporting the resolved node of the synthetic tree in the OToL website (<>). If the tree you are interested in updating is not in there, you can upload it via OToL's curator tool (<https://tree.opentreeoflife.org/curator).
- The alignment should be a gene alignment that was used to generate the tree. The original
  alignments are usually stored in a public repository such as TreeBase [@piel2009treebase; @vos2012nexml],
  DRYAD (<http://datadryad.org/>), or the journal were the tree was originally published.
  If the alignment is stored in TreeBase, `physcraper` can download it directly,
  either from the TreeBASE website (<https://treebase.org/>)
  or through the TreeBASE GitHub repository (SuperTreeBASE; <https://github.com/TreeBASE/supertreebase>).
  If the alignment is on another repository, or provided personally by the owner, a copy of it has to be
  downloaded by the user, and it's local path has to be provided as an argument.
- A taxon name matching step is performed to verify that all taxon names on the tips
  of the tree are in the DNA character matrix and vice versa.
- A ".csv" file with the summary of taxon name matching is produced for the user.
- Unmatched taxon names are dropped from both the tree and alignment.
  <!--The minimum amount of tips necessary for a physcraper run is just one!!!-->
  Technically, just one matching name is needed to perform the searches. Please, see next section.
- A ".tre" file and a ".fas" file containing only the matched taxa are generated and saved in the `inputs` folder to be used in the following steps.

## DNA sequence search and cleaning

- The next step is to identify the search taxon within the reference taxonomy. The search taxon will be used to constraint the DNA sequence search on the nucleotide database within that taxonomic group. Because we are using the NCBI nucleotide database, by default the reference taxonomy is the NCBI taxonomy. The search taxon can be provided by the user. If none is provided, then the search taxon is identified as the Most Recent Common Ancestor (MRCA) of the
  matched taxa belonging to the ingroup in the tree, that is also a named clade in the reference taxonomy. This is known as the Most Recent Common
  Ancestral Taxon (MRCAT; also referred in the literature as the Least Inclusive Common Ancestral Taxon - LICA).
  The MRCAT can be different from the phylogenetic MRCA when the latter is an unnamed clade in the reference taxonomy.
  To automatically identify the MRCAT of a group of taxon names, we make use of the OToL taxonomy tool (<https://github.com/OpenTreeOfLife/germinator/wiki/Taxonomy-API-v3#mrca>).

  Users can provide a search taxon that is either a more or a less inclusive
  clade relative to the ingroup of the original phylogeny. If the search taxon is more inclusive, the sequence search will be performed outside the MRCAT of the matched taxa, e.g., including all taxa within
  the family or the order that the ingroup belongs to. If the search taxon is a less inclusive clade, the users can focus on enriching a particular clade/region within the ingroup of the phylogeny.
- The Basic Local Alignment Search Tool, BLAST [@altschul1990basic; altschul1997gapped] is used to identify
  similarity between DNA sequences within the search taxon in a nucleotide
  database, and the accepted sequences on the alignment.
  The blastn function from the BLAST command line tools [@camacho2009blast] is used for local-database searches.
  A modified biopython blast function is used for web-based searches.
  <!--***How does the blast is setup? Is it an all-blast-all??***-->
- The DNA sequence similarity search can be done on a local database that is easily
  setup by the user. In this case, the blastn function is used to performs the similarity search [@camacho2009blast].
- The search can also be performed remotely, on the NCBI database. In this case, the
  bioPython BLAST function was modified to accepts is used to perform the similarity search.
- A pairwise alignment-against-all BLAST search is performed. This means that each sequence
  in the alignment is BLASTed against DNA sequences in a nucleotide database constrained to the search
  taxon. Results from each one of these BLAST runs are recorded, and matched sequences are saved
  along with their corresponding identification numbers (accesion numbers in the case of the GenBank database). This information will be used later to store the whole sequences in a dedicated library within the physcraper folder, allowing for secondary analyses to run significantly faster.
- Matched sequences below an e-value, percentage similarity, and outside a minimum
  and maximum length threshold are discarded. ***REPORT THE DEFAULT VALUES AND DESCRIBE WHAT THEY MEAN*** This filtering leaves out genomic sequences.
  All acepted sequences are asigned an internal identifier, and are further filtered.
- Because the original alignments usually lack database id numbers, a filtering
  step is needed. Accepted sequences that belong to the
  same taxon of the query sequence, and that are either identical or shorter than
  the original sequence are discarded. Only longer sequences belonging to the
  same taxon as the orignal sequence will be considered further for analysis.
- Among the remaining filtered sequences, there are usually several exemplars per taxon.
  Although it can be useful to keep some of them to, for example, investigate monophyly
  within species, there can be hundreds of exemplar sequences per taxon for some markers.
  To control the number of sequences per taxon in downstream analyses,
  5 sequences per taxon are chosen at random. This number is set by default but can be modified by the user.
- Reverse complement sequences are identified and translated.
- Users can choose to perform a more "cycles" of sequence similarity search, by blasting the newly found sequences. This can be done iteratively, but by default only sequences in the alignment are blasted. ***Is there an argument to control the number of cycles of blast searches with new sequences?***
- Accepted sequences are downloaded in full, and stored as a local database in a directory that is globally accesible (physcraper/taxonomy), so they are accesible for further runs.
- A fasta file containing all filtered and processed sequences resulting from the BLAST search is generated for the user.

## DNA sequence alignment

- The software MUSCLE [@edgar2004muscle] is implemented to perform alignments.
- First, all new sequences are aligned using default MUSCLE options.
- Then, a MUSCLE profile alignment is performed, in which the original alignment
  is used as a template to align new sequences. This ensures that the final alignment
  follows the homology criteria established by the original alignment.
- The final alignment is not further processed automatically. We encourage users
  to check it either by eye and perform manual refinement or using any of the many
  tools for alignment processing, to eliminate columns with no information.

## Tree reconstruction and comparison

- A gene tree is reconstructed for each alignment provided, using a Maximum Likelihood approach implemented with the software RAxML [@stamatakis2014raxml]
  with 100 classic rapid bootstrap [@felsenstein1985confidence] replicates by default. The number of bootsrap replicates can be modified by the user.
  Other type of bootstrap that I think is not yet incorporated into physcraper is the Transfer Bootstrap Expectation (TBE) recently proposed in @lemoine2018renewing.
- The original tree is used as starting tree for the ML searches. It can also be set as a full topological constraint or not used at all, depending on the goals of the user.
- Bootstrap results are summarized with Dendropy ADD CITATION
- The final result is an updated phylogenetic hypothesis for each of the genes provided in the alignment.
- Tips on all trees generated by physcraper are defined by a taxon name space, allowing to perform comparisons and conflict analyses.
- Robinson Foulds weighted and unweighted metrics ARE CALCULATED WITH DENDROPY TOO.
- Describe what a conflict analysis is: Node by node comparison between the original and updated tree, and the synthetic OToL using the conflict API of otol CITE REDELINGS AND HOLDER [@redelings and holder].

- For the conflict analysis to be meaningful, the root of the tree ineeds to be accurately defined.
- A SUGGESTED DEFAULT ROOTING BASED ON THE OPEN TREE TAXONOMY is implemented for now.  DESCRIBE HOW IT WORKS. SAY THAT IT IS A PROBLEM. Automatic rooting is not that smart yet. The best way right now is for users to define outgroups so trees are better rooted.
- Currently, the root is determined by finding the parent node of the sequences that do not belong to the ingroup/ search taxon. This ensures a correct rooting of the tree even when the search taxon is more inclusive than the ingroup.
- Conflict information can only be generated in the context of the whole Open Tree of
  Life. Otherwise, it is not really possible to get conflict data.
***- One way to compare two independent phylogenetic trees is to compare them both to
the synthetic OToL and then measure how well they do against each other***

# Examples

To exemplify the utility of physcraper we will address two use-case scenarios. One in which the user is interested in a particular group. Another one in which the user is interested in a particular phylogeny.
A tutrial as well as illustrated examples of steps needed to perform a physcraper analyses are available elsewhere.

## The hollies

A student is interested in the genus *Ilex*, the only extant clade within the family Aquifoliaceae, order Aquifoliales of flowering plants.
The genus encompasses between 400-600 living species. A review of literature reveals that there are three phylogenetic trees showing relationships within the hollies published.
The first one has been made available in TreeBASE as well as in the OToL phylesystem and is part of the synthetic tree. It samples 48 species.
The second tree has not been made available anywhere, not even in the supplementary data of the original publication.
<!-- ***Contact authors? They seem old school, probably do not wanna share their data.*** -->
The most recent one has been made available in the OToL Phylesystem and in the DRYAD repository. It is the best sampled yet, with 200 species. However, it has not been added to the syntehtic tree yet.
This makes it a perfect case to test the basic functionalities of physcraper: we know that the sequences of the most recently published tree have been made available on the GenBank database. Hence, we expect that updating the oldest tree will produce something very similar to the newest tree.

## The Malvaceae

A postdoc started working with a new reserach group. They are interested in solving relationships among lineages of the Malvaceae, a family of flowering plants with almost 6 000 known species.
A review of the literature shows them that there are many phylogenetic trees encompassing some of the linegaes in the group. However, the head of the group wants to use a particular marker they beilieve to be the best one to be able to solve the relationships in the group. They have been working in the alignment for long and they want to incorporate new data into the hypothesis of homology they have been curating and that they trust.

```{bash echo= FALSE, include=FALSE, message= FALSE, results='hide'}
wget "https://raw.githubusercontent.com/McTavishLab/physcraperex/master/vignettes/malvaceae-images/cotree-plot2-1.png"
mv cotree-plot2-1.png docs/figs/cotree-plot2-1.png
```
```{r fig-2, echo=FALSE, fig.cap="Comparison of original tree and update tree of the Malvaceae.", out.width = '85%', fig.retina= 2, fig.align = "center"}
knitr::include_graphics("docs/figs/cotree-plot2-1.png")
```
\newpage

# Discussion

Data repositories hold more information than meets the eye.
Besides the actual data, they have other types of information that can be used for the advantage of science.

Usually, initial ideas about the data are changed by analyses.
We expect that this new ideas on the data can be registered on data bases,
exposing new comers to expert understanding about the data.

There are many tools that are making use of DNA data repositories in different ways.
Most of them focus on efficient ways to mine the data -- getting the most homologs.
Some focus on accurate ways of mining the data - getting real and clean homologs.
Others focus on refinement of the alignment.
Most focus on generating full trees *de novo*, mainly for regions of the Tree of
Life that have no phylogenetic assessment yet in published studies, but also for
regions that have been already studied and that have phylogenetic data already.

All these tools are great efforts for advancing towards reproducibility in phylogenetics,
a field that has been largely recognised as somewhat artisanal.
We propose adding focus to other sources of information available from data repositories.
Taking advantage of public DNA data bases have been the main focus. However, phylogenetic knowledge is
also accumulating fast in public and open repositories.
In this way, the physcraper pipeline can be complemented with other tools that have
been developed for other purposes.

We emphasize that physcraper takes advantage of the knowledge and intuition of the expert
community to build upon phylogenetic knowledge, using not only data accumulated in
DNA repositories, but phylogenetic knowledge accumulated in tree repositories.
This might help generate new phylogenetic data. But physcraper does not seek to generate full phylogenies *de novo*.

Describe again statistics to compare phylogenies provided by physcraper via OpenTreeOfLife.
Mention statistics provided by other tools: PhyloExplorer [@ranwez2009phyloexplorer].
Compare and discuss.

How is physcraper already useful:
- to mine targeted sequences, in this way it is similar to baited analyses from PHLAWD and pyPHLAWD. Phylota does not do baited analyses, I think, only clustered analyses.
- Finding

How can it be used for the advantage of the field:
- rapid phylogenetic placing of newly discovered species, as mentioned in @webb2010biodiversity
- obtain trees for ecophylogenetic studies, as mentioned in @helmus2012phylogenetic
- one day could be used to sistematize nucleotide databases, such as Genbank [@benson2000genbank; @wheeler2000database], as mentioned in @san2010molecular, i.e., curate ncbi taxonomic assignations.
- allows to generate custom species trees for downstream analyses, as mentioned in @stoltzfus2013phylotastic


Things that physcraper does not do:
- analyse the whole GenBank database [@benson2000genbank; @wheeler2000database] to find homolog regions suitable to reconstruct phylogenies, as mentioned in @antonelli2017toward. There are already some very good tools that do that.
- provide basic statistics on data availability to assemble molecular datasets, as mentioned by @ranwez2009phyloexplorer. Phyloexplorer does this?
- it is not a tree repo, as phylota is, mentioned in @deepak2014evominer


# Acknowledgements

We acknowledge contributions from

The University of California, Merced cluster, MERCED (Multi-Environment Research Computer for Exploration and Discovery) supported by the National Science Foundation (Grant No. ACI-1429783).

# Authors' Contributions

# Data Avilability

# References
