TELL THEM WHAT YOU ARE GONNA TELL THEM
TELL TEHM
THEN TELL THEM WHAT YOU JUST TOLD THEM

However, if there are multiple alignments and trees, TreeBASE does not always indicate how they are connected. Hence, Phylesystem's metadata only provides a loose means of linking the tree with the exact alignment that generated it and a human curator must make the link.

over 217 million DNA sequences [(release number 238; June 15, 2020)](https://ftp.ncbi.nih.gov/genbank/gbrel.txt)
Data of number of nodes in the NCBI taxonomy can be found in their [homepage](https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=statistics&?&uncultured=hide&unspecified=hide&m=0)
In 2020 it has 668 929 taxa below spp, 474 760 species.

This in turns covers about 10% of extant described biodiversity [taking a conservative estimate of extant diversity; @scott2011ncbi; @federhen2003taxonomy].


Automated sequence gathering pipelines have been an important incorporation to the field of phylogenetics in many ways, particularly because they represent a clear step towards reproducibility in the field. In contrast, most published phylogenies to date have been inferred using alignments that have been assembled and curated "by hand" [@morrison2009would].

Automated approaches often suffer from errors in homology inference that could rapidly caught by human oversight. CITATION


The PHYLOTA [@sanderson2008phylota] tool has been used to construct homolog DNA datasets for about 42 studies reconstructing phylogenetic trees from various organisms.

In face of the science reproducibility crisis, more and more researchers are contributing to making data available, contributing to a continual growth in information available from different databases.

It is well known that GenBank [@benson2000genbank; @wheeler2000database] holds enormous amounts of genetic data, and it continues to grow. **Too clich√©**

A lot of this genetic data has the potential to be used to reconstruct the phylogenetic
history of various organisms [@sanderson2008phylota].
Pipelines that harness this potential have been available for over a decade now, such
as the Phylota browser, and PHLAWD. New ones keep on being developed, such as SUPERSMART
and the upgraded version of PHLAWD, PyPHLAWD.
Notably large phylogenies have been constructed using some of these tools,
Some others have not been used that much. So, how well accepted is this approach in the community?

Various efforts to automatize parts of the process in order to be able to take
advantage of data in public databases.

Most of this tools were envisioned for phylogenomics, were the analyses and processing of huge amounts of genetic data is naturally needed. Without automatization, curation of a phylogenomics database is a task that could not be finished in human time.
Some pipelines sacrificed precision and accuracy for quantity. But many tools to check
Examples of pipelines for phylogenomics:
BIR [Blast, Identify, Realign; @kumar2015bir] takes curated alignments as seed to find more data on genomic databases and create new phylogenomic hypotheses

Among other things, one of the major benefits stemming from these efforts are pipelines that make it easier to implement a reproducibile workflow in phylogenetic studies.


A widespread practice is a thorough visual inspection and manual curation of alignments, that are checked for "errors" that can usually result from similarity algorithms ***(give examples)***.
Commonly, alignment curators also take into account how the nature of the genetic marker, the taxon sampling and the organisms themselves, might affect the alignments. This hypotheses of homology are charged with expert knowledge, and hours of study, making them a scientifically reliable source if biological information.
As such, they represent an ideal platform to build up new phylogenetic knowledge.
Not many studies report its use.
Which might be an indication that the amount of manual effort that is required to curate gene alignments is still a bottleneck.


*** Introduce the Open Tree of Life***


In the past decade, various studies have developed solutions to automatize the first and second steps, by creating
pipelines that mine already available molecular data from the GenBank repository [@benson2000genbank; @wheeler2000database],
to obtain homologous characters that can be used for phylogenetic reconstruction.
These tools have been presented as aid for the nonspecialist to decrease some of
the difficulties in the generation of phylogenetic knowledge. However, they are
not that often used as so, suggesting that there are still difficulties for the nonspecialist.

The phylogenetic community has some reserves towards these tools, too. Mainly because they sometimes act as a black box.
However, automatizing the assembly of the character data set is a crucial step towards
reproducibility for a task that was otherwise primarily artisanal and hence largely
non-reproducible.

Phylogenies are important.

Generating phylogenies is not easy and it is largely artisanal. Although many efforts to automatize the process have been done, and the community is using those more and more, automatization of phylogenetic reconstruction is still not a widespread practice and among other benefits, it might be key for adoption of better reproducibility practices in the phylogenetics community. ***paragraph better to end discussion??? ***



The process of phylogenetic reconstruction implies many steps (that I generalize to the following):

1. Obtention of molecular or morphological character data -- get DNA from some organisms
and sequence it, or get it from an online nucleotide data repository, such as GenBank [@benson2000genbank; @wheeler2000database].
1. Assemble a hypothesis of homology -- Create a matrix of your character data, by
aligning the sequences, in the case of molecular data. Make sure thay are paralogs!
1. Analyse this hypothesis of homology to infer phylogenetic relationships among
the organisms you are studying -- Use different available programs to infer molecular
evolution, trees and times of divergence.
1. Discuss the inferred relationships in the context of previous hypothesis, the
biology and biogeography of the organisms, etc. -- Answer the question, *is this phylogenetic solution fair/reasonable?*

Each of these steps require different types of specialized training: in the field,
in the lab, in front of a computer, discussions with experts in the methods, and/or in the biological group of study.
All of these steps also require considerable amounts of time for training and implementation.

Even if it is hard to obtain phylogenies, we invest copious amounts of time and energy in generating them.
Phylogenies acan be used to solve issues such as food security, global warming, global health, all crucial for our well-being.
There is a lot of phylogenetic knowledge already available in published peer-reviewed studies.
In this sense, the non-specialists (and also the specialist) face a new problem: how do I choose the best phylogeny.

Public phylogenies can be updated with the ever increasing amount of genetic data that is available on GenBank [@benson2000genbank; @wheeler2000database].

The pipelines are so powerful and they will give you an answer, but there
is no way to assess if it is better than previous answers, it just assumes it is
better because it used more data.

A less automated workflow is one in which the alignments that generated the published
phylogeny are stored in other public database (such as DRYAD) or elsewhere (the users computer), and
are provided by the users.

A key aspect of the standard phylogenetic workflow is comparison with already existing
phylogenetic hypotheses and with phylogenies that are considered "best" by experts
not only in phylogenetics, but also experts on the focal group of study.

Physcraper uses the Open Tree of Life phylesystem and connects it
to the TreeBase database, to (1) get the original DNA data set matrices (alignments) that produced
a phylogeny that was published and then made available in the OToL database, (2)
use this DNA alignments as a starting point to get new genetic data belonging
to the focal group of study, to (3) finally update the phylogenetic relationships in the group.

On one hand, there are concerns about the use of automated pipelines in phylogenetics.
Concerns I think people have about these tools:
- Errors in identification of sequences
- Little control at different steps of the process
- Too much of a black box? Most of these phylogenies are being constructed by people learning about the methods (aka students), so they want to know what is going on at each step and end up doing it manually.

Many useful computational tools have been developed in an effort to make sense of the large amount of data in public molecular databases, as well as private ones. Generally referred to as "pipelines", most of these tools were motivated by the genomics revolution, to help researchers to deal with the massive amount of data, and automatically identify clusters of homologs for genomic assembly. Notably, this homolog DNA clusters can be used as homology hypotheses (i.e., molecular alignments) to reconstruct phylogenetic relationships.

Pipelines that automatize the assembly of DNA alignments from the GenBank database for phylogenetic reconstruction ("phylogenetic pipelines") such as PHYLOTA [@sanderson2008phylota], PHLAWD [@smith2009mega], and SUPERSMART [@antonelli2017toward], have been widely applied to study the evolutionary relationships among different organisms (TABLE?? maybe supplementary data), from a phylogeny of no more than 20 species of the family of barracuda fish [Sphyraenidae; @santini2013first], to a mega-phylogeny of almost 3 000 species of living ferns [Moniloformopses; @lehtonen2011towards].
