---
output:
  redoc::redoc
---

# Phylota overview

Phylota was published as a website to summarize and browse the phylogenetic potential of the GenBank
database [@sanderson2008phylota].

Since then, it has been cited 122 times for different reasons.


1. As an example of a tool that mines GenBank data for phylogenetic reconstruction,
or that is useful in any way for phylogenetics:
    - original publication of PHLAWD [@smith2009mega]
    - an analysis identifying research priorities and data requirements for resolving
    the red algal tree of life [@verbruggen2010data]
    - @beaulieu2012modeling cites phylota As an example study of very large and comprehensive
    phylogeny from mined DNA sequence data, (even if no phylogeny was really published
    there, only the method to do so)
    - a review for ecologists about phylogenetic tools [@roquet2013building]
    - a study constructing a dated seed plant phylogeny using pyPHLAWD [@smith2018constructing]
    - a study presenting an assembly and alignment free method for phylogenetic reconstruction
    using genomic data, that aims to be incorporated in a tool as phylota some day [@fan2015assembly].
    - nexml format presentation [@vos2012nexml] - cites phylota as a tool that uses
    stored phyloinformatic data that could benefit from adopting nexml, to increase
    interoperability.
    - a study of fruit evolution, analysing a previously published phylogeny of 8911
    tips of the Campanulidae, constructed with PHLAWD [@beaulieu2013fruit]
    - a study of Southeast Asia plant biodiversity inventory [@webb2010biodiversity] -
    cites phylota as a tool that would allow rapid phylogentic placing of newly
    discovered species, and generation of phylogenetically informed guides for field
    identification.
    - a study of wood density for carbon stock assessments [@flores2011estimating],
    cites phylota as an initiative to "get supertrees resolved up to species level".
    - a study proposing something similar to Open tree but applied only to land plants [@beaulieu2012synthesizing]
    - an analysis of the phylogenetic diversity-area curve [@helmus2012phylogenetic],
    cited phylota as a method alternative to phylomatic to "obtain plant phylogenetic
    trees for ecophylogenetic studies".
    - a study generating a phylogeny of 6,098 species of vascular plants from China
    [@chen2016tree] - uses DarwinTree [@meng2015darwintree] and generates sequence
    data *de novo* for 781 genera.
    - a review of the state of methods and knowledge generated by molecular systematics
    [@san2010molecular] cites phylota as a tool "intended to systematize GenBank information
    for large-scale molecular phylogenetics analysis".
    - the first phylotastic paper [@stoltzfus2013phylotastic] cites phylota as a "phylogeny
    related resource that provides ways to generate custom species trees for downstream use".
    - @antonelli2017toward cites phylota as a "pipeline that pre-processes entire GenBank
    releases in pursuit of sufficiently overlapping reciprocal BLAST hits, which are
    then clustered into candidate data sets". I also uses the PHYLOTA database in its
    own pipeline.
    - @deepak2014evominer present an algorithm for mining of frequent subtrees (common patterns)
    in collections of phylogenetic trees, as a way to extract meaningful phylogenetic
    information from collections of trees when compared to maximum agreement subtrees
    and majority-rule trees. They cite phylota as one of such tree collections available
    along with TreeBASE [@piel2009treebase].
    - @ranwez2009phyloexplorer cites phylota as a "program providing basic statistics
    on data availability for molecular datasets". They propose a tool to upload and
    explore user phylogenies to obtain detailed summary statistics on user tree collections.
    - @freyman2015sumac cites phylota as a tool that "provides a web interface to view
    all GenBank sequences within ta xonomic groups clustered into homologs" but that
    does not mine for targeted sequences, as opposed to NCBIminer or PHLAWD. They
    compare the performance of SUMAC to Phylota.
    - @chesters2013resolving cites phylota as a data mining tool that compiles metadata
    from mining of public DNA databases "for construction of large phylogenetic trees
    and multiple gene sets" and that the authors have recognised that gene annotations
    in public databases are insufficient and that careful partitioning of orthologous
    sequences is needed for supermatrix construction. @chesters2013resolving present
    a procedure that minimizes the problem of forming multilocus species units in
    a large phylogenetic data set using algorithms from graph theory.
1. When the software was actually used to construct (partially or in full) a DNA
data set to be used for phylogenetic reconstruction:
    - A 1000 tip phylogeny of the family of the nightshades [@sarkinen2013solanaceae]
    - A 56 tip phylogeny of crustacean zooplancton [@helmus2010communities] -- ecological study
    - A 63 tip phylogeny of the Salmonidae family [@crete2012salmonidae]
    - A 321 tip phylogeny of Testudines [@thomson2010sparse]
    - A 69 taxa phylogeny of the family Cyprinodontidae of the pupfish [@martin2011trophic]
    - A 2,957 taxa phylogeny of the class Moniloformopses of living ferns [@lehtonen2011towards]
    - A 2,573 species phylogeny of the Papilionoidea [@hardy2014specialization]
    - A 23 taxa phylogeny of the California flora [@anacker2011origins]
    - Phylogenies of 6 different clades of flowering plants representing an independent
    evolutionary origin of extrafloral nectaries: *Byttneria* (Malvaceae), *Pleopeltis* (Polypodiaceae),
    *Polygoneae* (Polygoneaceae), *Senna* (Fabaceae), *Turnera* (Passifloraceae), and *Viburnum*
    (Adoxaceae) [@weber2014defense].
    - To supplement DNA data sets of various pre-existing mammalian phylogenetic trees
    sampled at different taxonomic levels [@faurby2015species]
    - A 900 species tree of muroid rodents, Muroidea [@steppan2017muroid], where 300
    species were newly added by the study and the rest obtained using phylota.
    - A 95 taxa phylogeny of Gymnosperms, focused on Ephedra, Gnetales [@ickert2009fossil]
    - A 1061 genera phylogeny of the Oscine birds [@selvatti2015paleogene]
    - A 268 species phylogeny of sharks, representing all orders and 32 families [@sorenson2014effect]
    - A 466 species phylogeny of the Proteaceae, focusing on the species found in the Cape Floristic Region [@tucker2012incorporating].
    - A series of small phylogenies of unreported exact size, of sister groups of gall-forming insects [@hardy2010gall].
    - A 196 species phylogeny of the family Boraginaceae [@nazaire2012broad]. The authors
    actually found data for 318 Boraginaceae spp using phylota, but decided to reduce
    their data set to focus on the monophyly of genus *Mertensia*.
    - A phylogeny of 401 species of scale insects Coccoidea, Hemiptera [@ross2013large],
    with some sequences generated *de novo*.
    - Two phylogenies sampling all species of two different clades of insectivorous
    lizards, agamids and diplodactyline geckos, groups considered to be radiating
    in the Australiaâ€™s Great Victoria Desert [@rabosky2011species]
    - A phylogeny of 91 species of sparid and centracanthid fishes, Sparidae, Percomorpha,
    plus 2 outgroups, a lethrinid and a nemipterid exemplar [@santini2014first].
    - Updating a phylogeny of Arecaceae, constructing relationships in 6 cldes within
    the group: subfamilies Calamoideae and Coryphoideae, the tribe Ceroxyleae within
    subfamily Ceroxyloideae and three groups within subfamily Arecoideae: (1) Iriarteeae,
    (2) Cocoseae: Attaleinae except Beccariophoenix and (3) a group containing six
    tribes; Euterpeae, Leopoldinieae, Pelagodoxeae, Manicarieae, Geonomateae and Areceae
    [@faurby2016all].
    - A phylogeny of 768 Gesneriaceae species and 58 outgroups for a total species
    sampling of 826 taxa [@roalson2016distinct] some sequence were generated *de novo*.
    - A phylogeny of 47 species of scombrid fishes, with 2 outgroups, a gempylid and
    a trichiurid [@santini2013first].
    - to update a dataset underlying a large-scale fern phylogeny [@lehtonen2017environmentally],
    data set in <https://zenodo.org/record/345670#.Xr9QFRPYqqg>, also in TreeBASE,
    but it is one of those studies that is broken.
    - A phylogeny of 13 species of billfishes, order Istiophoriformes: Acanthomorpha,
    and four outgroups [@santini2013first]
    - A phylogeny of 765 aphid species, family Aphididae [@hardy2015evolution]
    - A phylogeny of less than 100 taxa of the family Ranunculaceae [@lehtonen2016sensitive],
    even though they retrieved info from phylota for 194 taxa within the family, they
    reduced their data set because of low sampling of markers for some taxa.
    - A phylogeny of 144 neobatrachian genera, assuming the monophyletic status of
    genera to increase matrix-filling levels [@frazao2015gondwana].
    - A 179 species phylogeny of the bird family Picidae (woodpeckers, piculets,
    and wrynecks) [@dufort2016augmented], augmented with data from an updated GenBank
    release and newly sequenced data.
    - A phylogeny of species of freshwater fish endemic to NorthAmerica [@strecker2014fish],
    phylota found data for 54 out of 66 spp.
    - A phylogeny of 520 species of the order Ericales [@hardy2012testing]
1. When the website was used to identify sequences and markers available in
GenBank for a particular group. In this cases, the dataset mining was either performed
with other tools, or not performed at all and just used for discussion:
    - A 812 tips phylogeny of the Order Chiroptera [@shi2015speciation] -- dataset
    constructed with PHLAWD
    - A 1276 tips phylogeny of the Fabaceae [@legume2013legume] -- dataset constructed
    by hand (I think??)
    - A review of dated phylogenies of fire-prone tropical savanna species from Brazil
    [@simon2012cerrado] -- just for discussion of the lack of markers available for
    these species on GenBank
    - A review of the phylogeetic sof the Apicomplexa, a parasitic phylum on unicellular
    protists [@morrison2009apicomplexa].
    - Three data sets from phylota (the suborder Pleurodira of side-necked turtles;
      the family Cactaceae of cacti; and the Amorpheae, a clade of legumes) were used
      to demonstrate and exemplify phylogenetic decisiveness [@sanderson2010phylogenomics]
1. Sometimes, it was cited by mistake:
    - In this 630 tip phylogeny of the Caryophyllaceae study [@greenberg2011caryophyllaceae] it might have been originally
    cited as an example of large phylogenies that reflect well supported relationships
    from previous smaller phylogenies. However, it was removed from the text but
    not from the final list of references. The DNA data set was constructed by hand
    most probably.
    - a study reconstructing the insect tree of life with 49,358 species, 13,865
    genera, and 760 families within the order Insecta [@chesters2017construction],
    uses its own algorithm (SOPHI) to mine public DNA databases [@chesters2014protocol].
    It does not cite phylota as it should, but includes it in their references.
1. Other miscellaneous uses of phylota:
    - @page2013bionames uses it to generate phylogenies for the [bionames website](http://bionames.org),
    a "database linking taxonomic names to their original descriptions, to taxa, and
    to phylogenies" generated with phylota.
