---
# Example from https://joss.readthedocs.io/en/latest/submitting.html
title: 'Physcraper: A Python package for continually updated gene trees'
tags:
  - Python
  - phylogenetics
  - update
  - species tree
  - Open Tree
authors:
  - name: Luna L. Sanchez Reyes
    orcid: 0000-0000-0000-0000
    affiliation: 1 # (Multiple affiliations must be quoted, like this: "1, 2")
  - name: Emily Jane McTavish
    orcid: 0000-0000-0000-0000
    affiliation: 1
affiliations:
 - name: University of California, Merced
   index: 1
 - name: Institution 2
   index: 2
citation_author: Sanchez-Reyes et. al.
date: 24 April 2020
year: 2020
bibliography: paper.bib
output: rticles::joss_article
csl: apa.csl
journal: JOSS
---

# Summary

Update trees that were reported on a published peer-reviewed study, uploaded to Treebase and to the Open Tree of Life database

# Intro

It is well known that GenBank holds enormous amounts of genetic data, and it continues to grow.
A lot of this genetic data has the potential to be used to reconstruct the phylogenetic history of various organisms [@sanderson2008phylota].
Pipelines that harness this potential have been available for over a decade now, such
as the Phylota brower, and PHLAWD. New ones keep on being developed, such as SUPERSMART
and the upgraded version of PHLAWD, PyPHLAWD.
Notably large phylogenies have been constructed using some of these tools,
Some other have not been used that much. So, how well accepted is this approach in the community?

Concerns:
Errors in identification of sequences
Little control along the process
Too much of a black box?

Most of these phylogenies are being constructed by people learning about the methods,
so they want to know what is going on.

The pipelines are so powerful and they will give you an answer, but there
is no way to assess if it is better than previous answers, it just assumes it is better because it used more data.

All these pipelines start tree consruction from zero?

The goal of Physcraper is to build upon previous phylogenegtic knowledge,
allowing a direct comparison of existing phylogenies to phylogenies constructed
using new information obtained from GenBank

To achieve this, Physcraper uses the Open Tree of Life phylesystem and connects it
to the TreeBase database, to use the original DNA dataset matrices that produced
a phylogeny in the OToL database as
a starting point to get new genetic data belonging to the focal group of study, and
update the phylogenetic relationships displayed by that dataset in particular.


Physcraper also allows node by node comparison of the trees, using the conflict API of OToL.

A less automated workflow is one in which the alignments that generated the published
phylogeny are stored in other public database (such as DRYAD) or elsewhere (the users computer), and
are provided by the users.

The original tree is used as starting tree for the phylogenetic searches, but it
can also be set as a full topological constraint or not used at all, depending on
the goals of the user.


# Examples

Updating trees from @schoch2009ascomycota

1. Get study id from Open Tree of Life. There are some options to do this:
  - You can go to the Open Tree of Life web browser, or
  - you can do it using R:
    - By getting the treebase ID of the study (which is not fully exposed on the Treebase website home page of the study, so you have to look it up manually)
```{r}
rotl::studies_find_studies(property = "treebaseId", value = "S2137")
```

1. Download the alignment from Treebase
If you are on the Treebase home page of the study, you can navigate to the matrix tab, and manually download the alignments that were used to reconstruct the trees reported on the study that were also uploaded to Treebase and to the Open Tree of Life repository.
  To make this task easier, you can use a command to download everything into your working folder:
```
auto_scrape.py -s pg_238 -t tree109 -o ../physcraper_example/treebase_alns
```
In this example, all alignments posted on Treebase were used to reconstruct both trees.

1. With the study id and the alignment files saved locally, we can do a physcraper run with the command:

```{bash}
opentree_scrape.py -s pg_238 -t tree109 -a treebase_alns/pg_238-LSU-M3956.nex -as "nexus" -o tmp
```


# Tools on a similar track:

Tools that do similar things:
pyPhlawd [@smith2019pyphlawd]
SUPERSMART [@antonelli]



# Acknowledgements

We acknowledge contributions from

# References
