---
title: 'Physcraper: A Python package for continually updated gene trees'
author:
  - Luna L. Sanchez Reyes
  - Martha Kandziora
  - Emily Jane McTavish
  - University of California, Merced
date: '`r format(Sys.time(), "%d %B, %Y")`'
bibliography: paper.bib
csl: apa.csl
output:
  redoc::redoc
---

# Abstract

1. Phylogenies are a key part of research in all areas of biology. Tools that automatize
some parts of the process of phylogenetic reconstruction (mainly character matrix construction)
have been developed for the advantage of both specialists in the field of phylogenetics and nonspecialists.
However, interpretation of results, comparison with previously available phylogenetic
hypotheses, and choosing of one phylogeny for downstream analyses and discussion still impose difficulties
to one that is not a specialist either on phylogenetic methods or on a particular group of study.

1. Physcraper is an open‚Äêsource, command-line Python program that automatizes the update of published
phylogenies by making use of public DNA sequence data and taxonomic information,
providing a framework for comparison of published phylogenies with their updated versions.

1. Physcraper can be used by the nonspecialist, as a tool to generate phylogenetic
hypothesis based on already available expert phylogenetic knowledge.
Phylogeneticists and group specialists will find it useful as a tool to facilitate comparison
of alternative phylogenetic hypotheses (topologies).
***Is physcraper intended for the nonspecialist?? We have two types of nonspecialists:
the ones that do not know about phylogenetic methods and the ones that might know
about phylogenetic methods but do not know much about a certain biological group.***

1. Physcraper implements node by node comparison of the the original and the updated
trees using the conflict API of OToL.

1. We hope the physcraper workflow demonstrates the benefits of opening results in phylogenetics and encourages researchers to strive for better data sharing practices.

1. Physcraper can be used with any OS. Detailed instructions for installation and
use are available at <https://github.com/McTavishLab/physcraper>.


# Introduction

Phylogenies are important.
Generating phylogenies is not easy.
The process of phylogenetic reconstruction implies many steps that can be generalized to the following:

1. Obtention of molecular or morphological character data -- get DNA from some organisms
and sequence it, or get it from an online repository, such as GenBank.
1. Assemble a hypothesis of homology -- Create a matrix of your character data, by
aligning the sequences, in the case of molecular data.
1. Analyse this hypothesis of homology to infer phylogenetic relationships among
the organisms you are studying -- Use different available programs to infer molecular
evolution, trees and times of divergence.
1. Discuss the inferred relationships in the context of previous hypothesis, the
biology and biogeography of the organisms, etc. -- Answer the question, *is this phylogenetic solution fair/reasonable?*

Each of these steps require different types of specialized training: in the field,
in the lab, in front of a computer, discussions with experts in the methods, and/or in the biological group of study.
All of these steps also require considerable amounts of time for training and implementation.

In the past decade, various studies have developed solutions to automatize the first and second steps, by creating
pipelines that mine already available molecular data from the GenBank repository,
to obtain homologous characters that can be used for phylogenetic reconstruction.
These tools have been presented as aid for the nonspecialist to decrease some of
the difficulties in the generation of phylogenetic knowledge. However, they are
not that often used as so, suggesting that there are still difficulties for the nonspecialist.
The phylogenetic community has some reserves towards these tools, too. Mainly because they sometimes act as a black box.
However, automatizing the assembly of the character data set is a crucial step towards
reproducibility for a task that was otherwise primarily artisanal and hence largely
non-reproducible.

Even if it is hard to obtain phylogenies, we invest copious amounts of time and energy in generating them.
They are crucial to solve problems such as food security, global warming, global health.
There is a lot of phylogenetic knowledge already available in published peer-reviewed studies.
In this sense, the non-specialists (and also the specialist) face a new problem: how do I choose the best phylogeny.

Public phylogenies can be updated with the ever increasing amount of genetic data that is available on GenBank.

A way to automatize the comparison of phylogenetic hypotheses and to allow reproducibility of the last step of the process.


A key aspect of the standard phylogenetic workflow is comparison with already existing
phylogenetic hypotheses and with phylogenies that are considered "best" by experts
not only in phylogenetics, but also experts on the focal group of study.

It is well known that GenBank holds enormous amounts of genetic data, and it continues to grow.
A lot of this genetic data has the potential to be used to reconstruct the phylogenetic
history of various organisms [@sanderson2008phylota].
Pipelines that harness this potential have been available for over a decade now, such
as the Phylota browser, and PHLAWD. New ones keep on being developed, such as SUPERSMART
and the upgraded version of PHLAWD, PyPHLAWD.
Notably large phylogenies have been constructed using some of these tools,
Some other have not been used that much. So, how well accepted is this approach in the community?

Concerns with these tools:
Errors in identification of sequences
Little control along the process
Too much of a black box?

Most of these phylogenies are being constructed by people learning about the methods,
so they want to know what is going on.

The pipelines are so powerful and they will give you an answer, but there
is no way to assess if it is better than previous answers, it just assumes it is
better because it used more data.

All these pipelines start tree construction from zero?

The goal of Physcraper is to build upon previous phylogenetic knowledge,
allowing a direct comparison of existing phylogenies to phylogenies that are constructed
using new genetic data from GenBank

To achieve this, Physcraper uses the Open Tree of Life phylesystem and connects it
to the TreeBase database, to (1) get the original DNA data set matrices (alignments) that produced
a phylogeny that was published and then made available in the OToL database, (2)
use this DNA alignments as a starting point to get new genetic data belonging
to the focal group of study, to (3) finally update the phylogenetic relationships in the group.

A less automated workflow is one in which the alignments that generated the published
phylogeny are stored in other public database (such as DRYAD) or elsewhere (the users computer), and
are provided by the users.

The original tree is by default used as starting tree for the phylogenetic searches, but it
can also be set as a full topological constraint or not used at all, depending on
the goals of the user.

Physcraper implements node by node comparison of the the original and the updated trees, using the conflict API of OToL.

# How does Physcraper work?

## The input: a study tree and an alignment

- The phylogenetic tree should be already in the Open Tree of Life store, or submitted
  via the curator system [@mctavish2015phylesystem].
  A user can choose from a variety of published trees supporting any node of the Tree of Life. If the tree you are
  interested in is not in Open Tree of Life, you can easily upload it via the curator
  tool (ADD URL).
- The alignment should be a gene alignment that was used to generate the tree. The
  alignments are usually stored in a public repository such as TreeBase [@piel2009treebase; @vos2012nexml],
  DRYAD (<http://datadryad.org/>), or the journal were the tree was originally published.
  If the alignment is stored in TreeBase, `physcraper` can download it directly
  either from the TreeBASE website (<https://treebase.org/>)
  or through TreeBASE GitHub repository, SuperTreeBASE (<https://github.com/TreeBASE/supertreebase>).
  If the alignment is on another repository, a copy of it has to be
  downloaded by the user, and it's local path has to be provided as an argument.
- A taxon name matching step is performed to verify that all taxon names on the tips
  of the tree are in the DNA character matrix and vice versa.
- A ".csv" file with the summary of taxon name matching is produced for the user.
- Unmatched taxon names are dropped from both the tree and alignment.
  <!--***What is the minimum amount of tips necessary for a physcraper run??*** Just one!-->
  Technically, just one matching name is needed to perform the searches. See below.
- A ".tre" and ".aln" files are generated and saved for a `physcraper` run.


## DNA sequence search and cleaning

- The next step is to identify and validate the search taxon. This must be a taxon (a named clade)
  from the NCBI taxonomy. It will be used to constraint the DNA sequence search on
  the GenBank database within that taxonomic group.
  By default, the search taxon is the most recent common ancestor (MRCA) of the
  matched taxa that is also a named clade in the NCBI taxonomy. This is refered to as the most recent common
  ancestral taxon (MRCAT) or the least inclusive common ancestral taxon (LICA).
  It can be different from the phylogenetic MRCA when the latter is an unnamed clade.
  This is done using the Open Tree API [taxonomy/mrca](https://github.com/OpenTreeOfLife/germinator/wiki/Taxonomy-API-v3#mrca).
  <!-- ***Does physcraper takes into account the focal clade from Open Tree in any way???***
  Not really, the focal clade is a search MRCA more than a taxonomic MRCA, physcraper's default is to get the ingroup from Otol, which is not the focal group.-->
  In the case that only one taxon is matched in both the tree and alignment, the MRCAT
  for that single taxon would be determined as HOW?,
  A search taxon can also be given by the user. It can be a more inclusive
  clade, if the user wants to perform a wider search, outside the MRCAT of the matched taxa, e.g., including all taxa within
  the family or the order. It can also be a less inclusive clade, if the user only
  wants to focus on enriching a particular clade/region within the tree.
- The BLAST algorithm is used to identify similarity between DNA sequences in a GenBank nucleotide
  database within the search taxon, and the remaining sequences on the alignment.
  <!--***How does the blast is setup? Is it an all-blast-all??***-->
- A pairwise all-against-all BLAST search is performed. This means that each sequence
  in the alignment is BLASTed against DNA sequences in a GenBank database constrained to the search
  taxon. Results from each one of these BLAST runs are recorded, and matched sequences are saved
  along with their corresponding GenBank accesion numbers. This information will be
  used later to store the whole sequences into a new library.
- The DNA sequence similarity search can be done on a local database that is easily
  setup by the user. In this case, the BLASTn algorithm is used to performs the similarity search.
- The search can also be performed remotely, on the NCBI database. In this case, the
  bioPython BLAST algorithm is used to perform the similarity search.
- Matched sequences below an e-value, percentage similarity, and outside a minimum
  and maximum length threshold are discarded. This filtering leaves out genomic sequences.
  All acepted sequences are asigned an internal identifier, and are further filtered.
- Because the original alignments usually do not have the GenBank accession numbers
  on the sequence names, a filtering process is needed. Accepted sequences that belong to the
  same taxon of the query sequence, and that are either identical or shorter than
  the original sequence are also discarded. Only longer sequences belonging to the
  same taxon as the orignal sequence will be considered for further analyses.
- Among the remaining filtered sequences, there are usually several exemplars per taxon.
  Although it can be useful to keep some of them to, for example, investigate monophyly
  within species, there can be hundreds of exemplar sequences per taxon for some markers.
  To control the number of sequences per taxon kept for further analyses, by default
  5 sequences per taxon are chosen at random. This number can be controlled by the user.
  <!--***How does pyphlawd, or phylota does the exemplar per sepcies choosing?***-->
- Reverse complement sequences are identified and translated.
- This cycle of sequence search is performed two times. ***Is there an argument to control the number of cycles of blast searches with new sequences***
- A fasta file containing all sequences resulting from the BLAST searches is generated for the user.

## DNA sequence alignment

- The software MUSCLE [@edgar2004muscle] is implemented to perform alignments.
- First, all new seweunces are aligned using default MUSCLE options.
- Then, a MUSCLE profile alignment is performed, in which the alignment of new sequences
  is aligned against the original alignment, working as a template. This ensures that
  the final alignment follows the homology criteria established by the original alignment.
- The final alignment is not further processed automatically. We encourage users
  to check it by eye and eliminate columns with no information. 

## Tree reconstruction

- A gene tree is reconstructed for each alignment provided, using RAxML with bootstrap replicates.
- The final result is a gene tree coupled to the conlict info.

## Tree comparison

- Conflict information can only be generated in the context of the whole Open Tree of
Life. Otherwise, it is not really possible to get conflict data.
***- One way to compare two independent phylogenetic trees is to compare them both to
the synthetic OToL and then measure how well they do against each other***

# Use case/ example

Imagine you are starting to work on a new biological group X. You have not much of an idea about
its phylogenetic relationships, you are a newly established researcher, and the group
is not anything any of your collaborators have worked on before.
A good idea is to start an intensive literature review on the phylogenetics of the group.
Rapidly, you find out there are 5 different phylogenies, that used different markers, and that
the papers, published at different times, do not discuss which phylogeny is
the one accepted by the expert community on X. You might need
to go to the annual conference of X, and even then, you might only find
different and contrasting opinions.
Somewhere along these months or even years doing this task, you looked into the
the OToL database. You found in there some or all the published trees of X, along
with a tree that has been deemed the best tree by curators and ideally experts on X?

## Ascomycota Example

Let's be more specific now about our X group and say it is the Ascomycota.
The best tree currently available in OToL was published by @schoch2009ascomycota.
The first step, is to get the Open Tree of Life study id. There are some options to do this:
- You can go to the Open Tree of Life website and browse until you find it, or
- you can get the study id using R tools:
  - By using the TreeBase ID of the study (which is not fully exposed on the
    TreeBase website home page of the study, so you have to really look it up manually):
```{r, treebase, eval = TRUE}
rotl::studies_find_studies(property = "treebaseId", value = "S2137")
```
  - By using the name of the focal clade of study (but this behaved very differently):
```{r, focal, eval = FALSE}
rotl::studies_find_studies(property="ot:focalCladeOTTTaxonName", value="Ascomycota")
```

Once we have the study id, we can gather the trees published on that study:
```{r, treeids, eval = TRUE}
rotl::get_tree_ids(rotl::get_study_meta("pg_238"))
rotl::candidate_for_synth(rotl::get_study_meta("pg_238"))
my_trees <- rotl::get_study("pg_238")
```
Both trees from this study have `r ape::Ntip(my_trees[[1]])` tips.

Let's check what one of the trees looks like:

<!-- ```{r fungi-phylo, eval = TRUE, fig.retina = 2, fig.width = 2} -->
<!-- ape::plot.phylo(ape::ladderize(my_trees[[1]]), type = "phylogram", cex = 0.1) -->
<!-- ``` -->

1. Download the alignment from TreeBase
If you are on the TreeBase home page of the study, you can navigate to the matrix tab, and manually download the alignments that were used to reconstruct the trees reported on the study that were also uploaded to TreeBase and to the Open Tree of Life repository.
To make this task easier, you can use a command to download everything into your working folder:
```
physcraper_run.py -s pg_238 -t tree109 -o ../physcraper_example/pg_238
```
In this example, all alignments posted on TreeBase were used to reconstruct both trees.

1. With the study id and the alignment files saved locally, we can do a physcraper run with the command:

```
physcraper_run.py -s pg_238 -t tree109 -a treebase_alns/pg_238tree109.aln -as "nexus" -o pg_238
```

## Testudines example

Phylogeny of the Testudines 6 tips from @crawford2012more
There is just one tree in OToL.
There is just one alignment on [treebase](https://treebase.org/treebase-web/search/study/matrices.html?id=12742) with all the 1 145 loci.


```
physcraper_run.py -s pg_2573 -t tree5959 -tb -db ~/branchinecta/local_blast_db/ -o pg_2573
```
# Discussion

Mention statistics provided by PhyloExplorer [@ranwez2009phyloexplorer]

## Tools that do similar things at different levels

PhyloFinder [@chen2008phylofinder] - a search engine for phylogenetic databases using
trees from TreeBASE - MEH

Phylota [@sanderson2008phylota] - cited by 122 studies.

PHLAWD [@smith2009mega] and pyPhlawd [@smith2019pyphlawd] - baited analyses

in @ceron2019phylotol: "pipelines such as AMPHORA (Wu and Eisen 2008) and Mega-phylogeny (Smith et al. 2009) focus on the construction and refinement of robust alignments rather than the collection of homologs"

PhyloExplorer [@ranwez2009phyloexplorer] - a python and MySQL based website to facilitate
assessment and management of phylogenetic tree collections. It provides "statistics describing the collection,
correcting invalid taxon names, extracting taxonomically relevant parts of the collection
using a dedicated query language, and identifying related trees in the TreeBASE database".

A [ruby pipeline](https://www.zfmk.de/en/research/research-centres-and-groups/taming-of-an-impossible-child-pipeline-tools-and-manuals),
only available from the [supplementary data](https://static-content.springer.com/esm/art%3A10.1186%2F1741-7007-9-55/MediaObjects/12915_2011_480_MOESM1_ESM.ZIP)
of the journal [[@peters2011taming]](https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-9-55#Sec21)

@chesters2014protocol presents an algorithm that mines GenBank data to delineate species in the insecta.
The authors present a nice comparison with the phylota algorithm.

PUmPER [@izquierdo2014pumper] - perpetual updating with newly added sequences to
GenBank

DarwinTree [@meng2015darwintree] predecessor is Phylogenetic Analysis of Land Plants Platform (PALPP) - takes data from GenBank, EMBL and DDBJ for land plants only.

NCBIminer [@xu2015ncbiminer]

SUMAC [@freyman2015sumac] -  both ‚Äúbaited‚Äù analyses and single‚Äêlinkage clustering
methods, as well as a novel means of determining when there are enough overlapping data in the DNA matrix

STBase - @mcmahon2015stbase present a pipeline for species tree construction and
the public database of one million precomputed species trees

@papadopoulou2015automated - Automated DNA-based plant identification for large-scale biodiversity assessment

SUPERSMART [@antonelli2017toward] - baited analyses up to bayesian divergence time
estimation

SOPHI - [@chesters2017construction] - Searches DNA sequence data from repos other
than GenBank, such as transcriptomic and barcoding repos.

OneTwoTree [@drori2018onetwotree] present a Web‚Äêbased, user-friendly, online tool for species-tree
reconstruction, based on the *supermatrix paradigm* and retrieves all available
sequence data from NCBI GenBank.

PhySpeTre [@fang2019physpetree] - no sequence retrieval, just phylogenetic reconstruction
pipeline.

Phylotol [@ceron2019phylotol] - "phylogenomic pipeline to allow easy incorporation
of data from  high-throughput sequencing studies, to automate production of both
multiple sequence alignments and gene trees, and to identify and remove contaminants.
PhyloToL is designed for phylogenomic analyses of diverse lineages across the tree
of life", i.e., bacteria and unicellular eukaryotes.


```{r phylota, child="phylota-overview.Rmd"}
```

# Acknowledgements

We acknowledge contributions from

# References
