%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% https://www.biomedcentral.com/getpublished                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% https://miktex.org/                                             %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
\usepackage{amsthm,amsmath}
%\RequirePackage[numbers]{natbib}
%\RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails

%\usepackage{hyperref}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\def\includegraphic{}
\def\includegraphics{}

%%% Put your definitions there:
\startlocaldefs
\endlocaldefs

%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Physcraper: A Python package for continually updated phylogenetic trees using the Open Tree of Life}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
  addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
  % corref={aff1},                       % id of corresponding address, if any
  email={sanchez.reyes.luna@gmail.com}   % email address
]{\inits{L.L.}\fnm{Luna L.} \snm{Sanchez Reyes}}
\author[
  addressref={aff1,aff2},
  email={kandziom@natur.cuni.cz}
]{\inits{M.}\fnm{Martha} \snm{Kandziora}}
\author[
  addressref={aff1},
  corref={aff1},
  email={ejmctavish@ucmerced.edu}
]{\inits{E.J.}\fnm{Emily Jane} \snm{McTavish}}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgdiv{School of Natural Sciences},             % department, if any
  \orgname{University of California},          % university, etc
  \city{Merced},                              % city
  \cny{USA}                                    % country
}
\address[id=aff2]{%
  \orgdiv{Department of Botany, Faculty of Science},
  \orgname{Charles University},
  %\street{},
  %\postcode{}
  \city{Prague},
  \cny{Czech Republic}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\begin{artnotes}
%%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
%\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                           %%
%% The Abstract begins here                  %%
%%                                           %%
%% Please refer to the Instructions for      %%
%% authors on https://www.biomedcentral.com/ %%
%% and include the section headings          %%
%% accordingly for your article type.        %%
%%                                           %%
%% I followed sections from here:            %%
%% https://bmcbioinformatics.biomedcentral.com/submission-guidelines/preparing-your-manuscript/software-article
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract}
\parttitle{Background}
Phylogenies are a key part of research in many areas of biology. Tools that automate
some parts of the process of phylogenetic reconstruction, mainly molecular character matrix assembly,
have been developed for the advantage of both specialists in the field of phylogenetics and non-specialists.
However, interpretation of results, comparison with previously available phylogenetic
hypotheses, and selection of one phylogeny for downstream analyses and discussion still impose difficulties
to one that is not a specialist either on phylogenetic methods or on a particular group of study.

\parttitle{Results}
Physcraper is a command-line Python program that automates the update of published
phylogenies by adding public DNA sequences to underlying alignments of previously
published phylogenies. It also provides a framework for straightforward comparison
of published phylogenies with their updated versions, by leveraging upon tools from
the Open Tree of Life project to link taxonomic information across databases.
The program can be used by the nonspecialist, as a tool to generate phylogenetic
hypotheses based on publicly available expert phylogenetic knowledge.
Phylogeneticists and taxonomic group specialists will find it useful as a tool
to facilitate molecular dataset gathering and comparison
of alternative phylogenetic hypotheses (topologies).

\parttitle{Conclusions}
The Physcraper workflow showcases the benefits of doing open science for
phylogenetics, encouraging researchers to strive for better sharing practices.
Physcraper can be used with any OS and is released under an open-source license.
Detailed instructions for installation and
usage are available at \href{https://physcraper.readthedocs.io}{https://physcraper.readthedocs.io}.


\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{gene tree}
\kwd{gene phylogeny}
\kwd{multilocus}
\kwd{interoperability}
\kwd{open science}
\kwd{reproducibility}
\kwd{public database}
\kwd{DNA alignment}
\kwd{Open Tree of Life}
\kwd{otol}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for two column layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                            %%
%% The Main Body begins here                  %%
%%                                            %%
%% Please refer to the instructions for       %%
%% authors on:                                %%
%% https://www.biomedcentral.com/getpublished %%
%% and include the section headings           %%
%% accordingly for your article type.         %%
%%                                            %%
%% See the Results and Discussion section     %%
%% for details on how to create sub-sections  %%
%%                                            %%
%% use \cite{...} to cite references          %%
%%  \cite{koon} and                           %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}      %%
%%                                            %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
Phylogenies capture the shared history of organisms and provide key evolutionary
context for our biological observations \cite{dobzhansky1973nothing}.
Updating existing phylogenies with publicly available molecular sequence data provides
the opportunity to simultaneously study
the evolutionary history of many taxa in a reproducible and continuous manner.
Here, we introduce Physcraper, a tool that establishes a data interoperability
framework for biological databases to automate data connections across biological databases.
Physcraper's main goal is to build upon published alignments and published phylogenies
to extend existing phylogenetic inferences with more data and taxa, which improve
phylogenetic reconstruction \cite{hillis1996inferring, natsidis2019phylogenomics},
time of divergence estimates \cite{schulte2013undersampling, soares2015influence},
biogeographic analyses \cite{kayaalp2017back},
as well as help in resolving phylogenetic conflict \cite{hedtke2006resolution, townsend2010optimal, natsidis2019phylogenomics}.
Physcraper updates a starting single locus alignment and corresponding phylogeny
with public DNA data from GenBank \cite{benson2000genbank}, and links the tips in the updated trees to a unified, interoperable
taxonomic resource \cite{rees2017automated}.

Data such as
geographical location, fossil ranges, and genetic and phenotypic information increasingly
available in public biological databases constitute an amazing
resource for scientific discovery \cite{baxevanis2015importance}.
One of the main challenges for automatic
integration of data across biological databases are varying taxonomic idiosyncrasies.
To address this challenge, the Open Tree of Life project (OpenTree)
created a unified taxonomy for automatic taxonomic name standardization, by integrating taxonomic
data from several databases \cite{rees2017automated}, including the USA National
Center for Biodiversity Information (NCBI) taxonomy \cite{federhen_ncbi_2012, schoch_ncbi_2020},
and the Global Biodiversity Information Facility (GBIF) \cite{gbif_secretariat_gbif_2019}, among many others.
All of OpenTree tools and methods are carried out via OpenTree's Application Programming Interfaces (APIs),
that are open as services for use by the general public \cite{opentreeAPIs}.
Physcraper leverages on existing OpenTree's taxonomic APIs to
standardize taxon names in any phylogeny, providing
a framework for automatic connection of updated phylogenies with data from any biological database.

Decades of single locus sequencing have generated
massive amounts of homologous DNA datasets that have the potential to be used for phylogenetic
reconstruction at many scales \cite{sanderson2008phylota}.
More than a decade ago, GenBank release 159 (April 15, 2007) already hosted 72
million DNA sequences that were gauged to have the potential to resolve phylogenetic
relationships of 98.05\% of the almost 241,000
distinct taxa in the NCBI taxonomy at the time \cite{sanderson2008phylota}.
However, even thirteen years later, phylogenetic estimates for most of these taxa
are still not available \cite{mctavish_how_2017}.
Using its unified taxonomy, OpenTree assembles a comprehensive synthetic tree of life comprising
2.3 million tips, of which around only
90,000 are supported by phylogenetic data uploaded to OpenTrees' database (the Phylesystem
\cite{mctavish2015phylesystem}) by curators - the remaining 1.4 million taxa are placed
in the tree based on taxonomy.
There is a considerable amount of phylogenetically
informative data in GenBank with the potential to fill these phylogenetic gaps in
the tree of life, but this data either has not been analysed or the analyses
have not been made publicly available \cite{mctavish_how_2017}.

Assembling a DNA alignment from a massive database such as GenBank can be done ``by hand",
but that is a time-consuming approach which is largely not reproducible.
Various computational pipelines that mine DNA databases fast, efficiently, and reproducibly
have been developed and used to infer phylogenetic relationships of many organisms
(e.g., \cite{smith2009mega, antonelli2017toward, izquierdo2014pumper, pearse2013phylogenerator}).
While genomics has, and will continue to, revolutionize phylogenetic inference,
the diversity of alternative genomic sequencing approaches that are implemented produce
widely non-overlapping homology hypotheses across taxa \cite{jones2016targeted},
creating challenges for phylogenetic reconstruction.
Phylogenomics addresses this problem by focusing on targeted capture of informative
regions \cite{andermann2020guide}.
However, fine-grained curated markers and alignments can significantly improve phylogenetic
reconstructions, even in phylogenomic analyses \cite{fragoso2017pilot}.

Physcraper improves on previous work that automates phylogenetic reconstruction, by leveraging
on the knowledge contained in existing homology hypotheses that phylogeneticists and taxon specialists have assessed
and deemed appropriate for a specific phylogenetic scope.
There are almost 8,200 publicly available, peer-reviewed curated alignments, covering
around 100,000 distinct taxa in the TreeBASE database
\cite{piel2009treebase, vos2012nexml, piel2018treebasedmp},
which can be used as seeds to mine molecular databases, and as ``jump-start'' alignments
for phylogenetic reconstructions \cite{morrison2006multiple} to continually enrich,
update and compare new phylogenetic hypotheses to existing knowledge.

Physcraper is implemented as a Python pipeline that uses OpenTree's APIs to automatically
link any phylogeny mapped to OpenTree's standardized taxonomy,
to alignments from TreeBASE, and data from GenBank.
Physcraper's usage and functionalities are presented with a case-study analysis of a
group of flowering plants, the hollies.


\section*{Implementation}

Physcraper is implemented with Python and can be run on a Python interactive session,
as a Python script, or using the command line interface we developed for it.
It currently consists of 13 modules.
%% , classes, methods and functions.%%
For testing and improving Physcraper's Python code syntax quality, we
used the Pylint software following instructions from its website \cite{pylintWeb}
 and manual \cite{pylintManual}, with a ``.pylintrc" configuration file.

We improved code syntax of Physcraper's modules with low Pylint scores, and fixed code errors by
following Pylint's recommendations.
Based on Physcraper's software design choices, some of Pylint's recommendations were overruled
by using its check-disabling system, and are explained along the code.
As of now, all Physcraper modules have a Pylint score of 10/10.

% Pylint score for each module and locally disabled Pylint reccommendations
% physcraper/aligntreetax.py		10/10	pylint disabling “too many branches” and “too many locals”
% physcraper/configobj.py 		10/10  pylint global disabling attributes defined outside init
% physcraper/helpers.py		10/10
% physcraper/ids.py			10/10 	pylint disabled unused-variables that are used later by other functions
% physcraper/ncbi_data_parser.py	10/10
% physcraper/opentree_helpers.py 10/10		we pylint-disabled “too many local variables”, “too many branches” and “too many statements”
% physcraper/scrape.py		10/10	commented out a bare except and pylint-disabled ”globals”, “too many public methods”, “too many local variables”, “too many lines in module”, “too many branches”, “too many statements” and “no self use”
% physcraper/treetaxon.py		10/10
% bin/find_trees.py			10/10
% bin/physcraper.py			10/10
% bin/tree_comparison.py		10/10
% bin/multi_loci.py			10/10 		Global disabled “redefined-outer-name”, “protected-access”, “too-many-branches”, “too-many-statements”, “too-many-locals”

The general Physcraper framework (Figure 1) consists of 4 steps: 1) identifying and
processing a phylogeny and its underlying alignment; 2) performing a BLAST search of
DNA sequences from original alignment on GenBank, and filtering of new sequences;
3) profile-aligning new sequences to original alignment; 4) performing a phylogenetic
analysis and comparing the updated results to existing phylogenies.


\subsection*{The inputs: a phylogeny and an alignment}

Taxon names in the input phylogeny must be standardized to the OpenTree taxonomy
\cite{ott3.2} using OpenTree's bulk Taxonomic Name Resolution Service
(TNRS) tool \cite{TNRStool}. Users can upload their
own phylogeny, or choose from among the 2, 950 curated phylogenies stored in OpenTree's
Phylesystem database \cite{phylesystemGithub} that also have
alignments available on the TreeBASE database \cite{treebase_website, supertreebase}.

The input alignment is a single locus DNA dataset that was used in part or in
whole to generate the input phylogeny. Physcraper retrieves TreeBASE alignments
automatically. Alternatively, users can provide the path to a local copy of the
alignment of their choosing.
Only taxa that are both in the sequence alignment and in the phylogeny are considered
further for analysis; at least one taxon and its corresponding sequence are required.

\subsection*{DNA sequence search and filtering}

The Basic Local Alignment Search Tool, BLAST \cite{altschul1990basic} is used for DNA
sequence search on a remote or local GenBank database. It is constrained to a
 ``search taxon", which corresponds to the Most Recent Common Ancestor (MRCA) of
all ingroup taxa that is also a named clade in the NCBI taxonomy (Figure 1). The
search taxon is identified using OpenTree's taxonomic API \cite{mrcaAPI}.

BLAST is performed using the blastn algorithm \cite{camacho2009blast}
implemented in BioPython 1.71 \cite{cock2009biopython} NCBIWWW module \cite{ncbiwww}
modified to accept an alternative BLAST address.
Each sequence in the alignment is BLASTed once against all DNA sequences in GenBank.
New sequences are excluded for analysis if they 1) are not in the search taxon;
2) have an e-value above the cutoff (default to 0.00001); 3) fall outside a minimum
and maximum sequence length threshold, defined as a proportion of the average sequence length without
gaps of all sequences in the input alignment (default values of 80\% and 120\%,
respectively); 4) or, if they are either identical to or shorter than an existing
sequence in the input alignment and they represent the same taxon in OpenTree's or
NCBI taxonomy.
An arbitrary maximum number of randomly chosen sequences per taxon are allowed
(default to 5).

Reverse, complement, and reverse-complement sequences are identified and translated
using BioPython internal functions \cite{cock2009biopython}.
Iterative cycles of BLAST searches can be performed, by blasting all new sequences
until no new ones are found. By default only one BLAST cycle is performed.

\subsection*{New DNA sequence alignment}

MUSCLE \cite{edgar2004muscle} is used to perform a profile alignment in which the
original alignment is used as a template of homology criteria to align new sequences.
The final alignment is not further automatically checked, and additional inspection
and refinement are recommended.

\subsection*{Phylogenetic reconstruction and comparison}

RAxML \cite{stamatakis2014raxml} is implemented to reconstruct a Maximum Likelihood
(ML) gene phylogeny for each input alignment with default settings (GTRCAT model and
100 bootstrap replicates with default algorithm), using the input phylogeny as starting
tree for ML searches.
Bootstrap results are summarized using DendroPy's SumTrees module
\cite{sukumaran2010dendropy}.

Physcraper's main result is an updated phylogenetic hypothesis for the search taxon.
Updated and original phylpogeny are compared with Robinson-Foulds weighted and unweighted
metrics calculated with Dendropy \cite{sukumaran2010dendropy}, and with a node by node
comparison between the synthetic OpenTree and the original and updated phylogenies individually,
using OpenTree's conflict API \cite{redelings2017supertree}.


\section*{Results}
\subsection*{Case Study: The hollies}

A user is interested in phylogenetic relationships within the genus \textit{Ilex}. Commonly
known as ``hollies", the genus encompasses between 400 \cite{gottlieb2005molecular}
and 500 recognized living species \cite{plantlist}, and is
the only extant taxon within the family Aquifoliaceae, in the order Aquifoliales of flowering
plants \cite{chase2016update}.

An online literature review in June 2020 (Google scholar search for ``ilex phylogeny")
revealed that there are several published studies addressing phylogenetic relationships within
the hollies \cite{cuenoud2000molecular, gottlieb2005molecular, manen2010history,
setoguchi2000intersectional, selbach2009new, yao2020phylogeny}, but only the ``Gottlieb2005"
study \cite{gottlieb2005molecular} and the ``Yao2020" study \cite{yao2020phylogeny}
have data openly available.
The Gottlieb2005 phylogeny and alignment are available in TreeBASE study 1091 \cite{treebase1091}.
The Gottlieb2005 phylogeny samples 41 species, is available in
OpenTree's Phylesystem (study pg\_2827 \cite{pg2827}),
and has been integrated into OpenTree's synthetic tree \cite{mrcaott68451ott89474}.
The Yao2020 \textit{Ilex} phylogeny is the most recent one for the genus \cite{yao2020phylogeny},
and it is only available in OpenTree's Phylesystem (study ot\_1984 \cite{ot1984}),
and in the DRYAD repository \cite{yao2020dryad}.
With 175 tips, the Yao2020 phylogeny \cite{yao2020phylogeny} is the best sampled phylogeny available for
the genus \textit{Ilex}.
In order to showcase Physcraper's performance, we chose the Gottlieb2005 phylogeny
and a corresponding single locus alignment of the internal transcribed
spacer DNA region (ITS) as Physcraper inputs, to update relationships in the genus \textit{Ilex}.
Currently being the best sampled and most recent phylogenetic hypothesis for \textit{Ilex},
we used the Yao2020 phylogeny as the ideal standard to compare results from this
Physcraper case study.

We ran Physcraper on a computer node of the Multi-Environment Research Computer for
Exploration and Discovery (MERCED) cluster from the University of California,
Merced, using one core at 2301 MHz RAM. BLAST and RAxML analyses ran for 19hrs 45min, with bootstrap
analyses taking an additional 13hrs.
%% <!-- MTH: you probably need some details about the hardware, given the fact that you are discussing running times -->
The Gottlieb2005 phylogeny \cite{gottlieb2005molecular} updated using Physcraper (Figure 2;
Physcraper updated phylogeny from now on) displays all 41 distinct
taxa from the original study plus 231 new tips, contributing phylogenetic data to
84 additional \textit{Ilex} taxa. The best ML phylogeny from the RAxML analysis is 99\%
resolved, with 25\% of nodes with bootstrap support $<$ 0.1 and 48\% nodes with
bootstrap support $>$ 0.75 (figure 2A).
A large portion of internal branches are negligibly small, with 30 branches
$<$ 0.00001 substitution rate units, from which only 9 have a bootstrap support $>$ 0.75
(Figure 2).
As comparison with the Physcraper updated phylogeny, the Yao2020 phylogeny \cite{yao2020phylogeny}
also contains all 41 distinct taxa
sampled in the Gottlieb2005 phylogeny \cite{gottlieb2005molecular},
while contributing phylogenetic data to 134 additional \textit{Ilex} taxa. From these,
67 taxa are also in the Physcraper updated phylogeny.
While the Yao2020 phylogeny \cite{yao2020phylogeny} was also constructed using
ITS as a marker, their GenBank data is not released yet. Hence, Physcraper was unable
to incorporate 68 taxa that are only on the Yao2020 phylogeny because the DNA data is unavailable.
We also note that Physcraper incorporates 18 \textit{Ilex} taxa that are not in the Yao2020
phylogeny \cite{yao2020phylogeny}. These taxa appear nested among other \textit{Ilex}
species (Figure 2B) and visual inspection of the DNA sequences suggests they are
correctly assigned as \textit{Ilex}.
The ITS alignment that underlies the Yao2020 phylogeny was constructed without any tool to
mine GenBank \cite{yao2020phylogeny}, which could explain why Physcraper was able to
incorporate these 18 additional \textit{Ilex} taxa in the Physcraper updated phylogeny.

%% Maybe show conflict results? and then add to the discussion. YES

\subsection*{Verification test}

To test the performance of Physcraper we performed a verification test in which
we pruned 9 out of the 41 tips in the original Gottlieb2005 phylogeny \cite{gottlieb2005molecular},
corresponding to a 20\% trim, excluding the outgroups. We then performed a Physcraper
run to test if we would recover the pruned tips. To perform the BLAST searches,
Physcraper automatically drops sequences from the alignment belonging to taxa
absent from the input phylogeny.
The Physcraper updated phylogeny successfully recovered only 6 out of 9 pruned tips.
Closer examination of results revealed that sequences for the 3 missing tips were correctly retrieved
with BLAST along with the 6 sequences belonging to the remaining pruned tips.
We then followed the GenBank accession numbers reported in \cite{gottlieb2005molecular}
belonging to the sequences of the 3 missing tips, \textit{Ilex warburgii} (accession
number reported: U92600/U92601; updated: AH007153.2), \textit{Ilex dimorphophylla}
(accession number reported: U92592/U92593; updated: AH007149.2),
and \textit{Ilex percoriacea} (accession number: AH007156.2). We note that these three sequences
contain a 100 bp long gap of unidentified nucleotides (Ns) that is completely absent
from the original alignment. This caused them to exceed Physcraper's default sequence
length cutoff of 120\%, being thus filtered and excluded from further analyses.
The missing ITS sequences do appear in Physcraper's output file ``seqlen\_mismatch.txt",
that includes the accession number, taxon name, and sequence length of all BLAST matches
that were filtered based on sequence length cutoffs set in the configuration file.


\section*{Discussion}
Databases preserving and democratizing access to biological data
have become essential resources for science.
New molecular data keep accumulating and tools facilitating its integration into
existent evolutionary knowledge contribute to the acceleration of scientific discovery.

Physcraper is a tool that builds upon previous knowledge stored in published
alignments and phylogenies, taking advantage of OpenTree's services to facilitate comparison
of phylogenies, with the main goal of extending our knowledge of phylogenetic relationships
across the tree of life.

We believe this is a key step to successfully establish an open, reproducible workflow for phylogenetics,
facilitating phylogenetic knowledge for ecologists and other non-specialists, effectively
democratizing phylogenetic studies.

As a tool for automatizing phylogenetic reconstruction from molecular databases,
Physcraper presents several advantages over existing phylogenetic pipelines designed to
make evolutionary sense of the vast amount of
public molecular data available.

Several analysis tools create full phylogenies \textit{de novo} by mining of molecular
databases \cite{antonelli2017toward, sanderson2008phylota, smith2019pyphlawd,
bennett2018phylotar, pearse2013phylogenerator}.
In particular, Phylota \cite{sanderson2008phylota},
and PHLAWD \cite{smith2009mega}, have been cited and used abundantly.

Physcraper builds on this automated database mining concept by incorporating prior
phylogenetic work and existing taxonomic domain knowledge on appropriate markers
and alignment construction.
This decreases error (requiring less manual downstream processing) and
eases comparison with previous phylogenetic knowledge.

Results from the verification test highlight the importance of incorporating existing expertly
curated homology statements to automatically update phylogenetic relationships, instead
of ignoring the information they contain and building homology statements fully \textit{de novo}.

We encourage users to look at the output files containing information about the
filtered sequences, and potentially modify configuration parameters such as the
sequence length cutoff parameter, based on the filtered sequences.
Default filtering parameters are arbitrary, but we hope that by making the process
of locating homologous sequences online reproducible, and tracking what filters
are used, we make it easier for researchers to delve into the effect that different
choices have on their inferences.
This is in contrast to ``manual" searches for taxa, where similarly arbitrary filters
are applied, but are difficult to trace.
As many studies have shown \cite{huang_what_2009} the effect of missing data can
be enigmatic, and interact with the true phylogenetic relationships for the data
set at hand.
There is not currently strong support in the literature for any particular cutoff
value, and rather than prescribe specific approaches, we encourage users to explore
the effects of different choices on their phylogenetic inferences.
In addition, by providing the output files at each step of the analysis,
it is straightforward to assess how changing parameter and software choices do or
do not drive differences in phylogenetic inference.
By gathering the sequences, and making the unaligned files easily available to users,
researchers can compare if applying any alternate alignment tool of their choice
affects inferences.
Once sequences are aligned, they can apply and compare inferences from any phylogenetic
software.

Organellar genome sequences, such as chloroplasts and mitochondria will also generally
be excluded from automatic addition based on default Physcraper length cutoffs.
Multiple sequence alignment of loci of drastically different lengths is unfeasible,
and we have found in testing that it often returns incorrect results, splitting
shorter sequences with many long gaps to align with exact matches across the entire
longer locus.
While it would be possible to directly extract the BLAST match from genomes, this
 would exclude potentially homologous flanking regions which are not matched
by BLAST's local search algorithm, but that may be important for phylogenetic inference.
Instead we list the accession numbers for these matches in the ``seqlen\_mismatch.txt"
file, for users to assess and incorporate appropriate homologous regions to their
alignment of interest.

Unlike phylogenetic placement approaches \cite{berger_performance_2011, matsen_pplacer_2010},
which add new taxa without modifying the input phylogeny, Physcraper estimates all
the relationships anew in the context of the new data.
PUMPER \cite{izquierdo2014pumper} shares these conceptual strengths, but is no
longer under active development, is challenging to install and run, and has resulted
in very few phylogenetic analyses since its publication.

Physcraper generates gene trees, which individually do not capture the full complexity
of species' evolutionary history \cite{song2012resolving}. In addition, single gene
phylogenies with very high numbers of taxa may lack sufficient signal
for accurate phylogenetic resolution \cite{morel_phylogenetic_2020}.
The Physcraper workflow avoids this challenge by focusing on ingroup
taxa of an existing phylogeny, using markers that have been assessed and proven
appropriate for that phylogenetic scope in past publications.
Also, Physcraper thins alignments by removing sequences identical to original
and newly added sequences, and by setting a maximum number of sequences per taxon.
Nonetheless, it is incumbent on users to assess their final inference with respect
to statistical support and biological plausibility.

In the era of phylogenomics, rigorous analyses of multiple loci still allow for
more complex evolutionary models
than analyses of large genomic data sets, and in many cases can provide better evolutionary estimates.
For example \cite{zhu_complexity_2021} show that when applying coalescent models,
there is more information in two genes of 300 bp each than in 600 independent sites.
Physcraper is designed to facilitate
gathering alignments and gene trees for multiple loci from a group of interest,
that together can be used to reconstruct species trees taking into account coalescent
processes with ASTRAL \cite{mirarab2014astral} or SVD Quartets \cite{chifman2014quartet}.
Physcraper's ``multi\_locus.py'' module allows to automatically
merge the outputs of Physcraper runs from different loci into input files for the
two software mentioned above, or as concatenated alignments for supermatrix analyses.


%% Biological plausibility of example??
Our example application of Physcraper to update a phylogeny of the genus \textit{Ilex} is based on a
single marker, so we expect for it to be not as well resolved as phylogenies resulting from analyses
that used multiple markers. Although not perfect,
we think the Physcraper updated \textit{Ilex} phylogeny seems biologically reasonable
in different ways. All samples
corresponding to the ingroup cluster together forming a monophyletic group (Figure 2A), and
samples belonging to the same \textit{Ilex} species also form
monophyletic groups (Figure 2B). A notable exception is samples of the species \textit{Ilex theeizans}, which appear as
non-monophyletic in the updated phylogeny as well as in the original Gottlieb2005 phylogeny.
%This suggests that the samples might represent two different species, but further
analyses should be conducted
A visual comparison of the Yao2020 phylogeny and the original Gottlieb2005 phylogeny suggests
that the relationships within the genus \textit{Ilex} are still being actively
determined, and that increased taxon sampling might be key to resolve them.

Physcraper has the added advantage of facilitating the linkage of taxonomic information about tips
in the output phylogenies to data available in a variety of biological databases
\cite{rees2017automated}, such as
geographical locations for taxa from the GBIF \cite{gbif_secretariat_gbif_2019}.
Taxonomic links, and comparisons to existing published phylogenies in the OpenTree
data store can also help flag paralogous sequences.
Accidentally including paralogs as homologs is a risk in phylogenetic analyses,
and can be more prevalent in automated analyses than in manually curated analyses.
We provide users with several tools to try to assess homology of their aligned sequences.
The estimated gene phylogeny itself is an evolutionarily explicit way to visualize gene
evolution, which in concert with taxonomic labelling can reveal paralogy.
OpenTree's conflict analysis tool informs the users of whether their phylogeny contains major
conflicts with established taxonomy and any phylogenetic context they wish to compare to.
This tool also returns information on taxonomic and phylogenetic conflicts that exist
in the original input phylogeny.
Detected conflicts may be a sign that taxonomy needs to be updated, or may be a sign
that non-homologous sequences have been included in the analysis.
These taxonomic and phylogenetic conflicts flag regions of the phylogeny for the researcher
to more closely examine and assess homology.

The Physcraper workflow can be used to rapidly (in a matter of hours) create
phylogenies which can address challenges overarching both fields of ecology and evolution, such as
phylogenetically placing newly discovered species \cite{webb2010biodiversity},
curating taxonomic assignments
\cite{san2010molecular},
and generating custom trees for ecological \cite{helmus2012phylogenetic} and
evolutionary downstream analyses \cite{stoltzfus2013phylotastic}.

\section*{Conclusions}
Data repositories hold more information than meets the eye.
Beyond the main data, they are rich sources of metadata that can be leveraged
for the advantage of all areas of biology as well as the advancement of scientific
policy, applications and education.
Initial ideas about the data are constantly changed by results from new analyses.
Physcraper provides a framework for reproducible phylogenetics that has the
potential to consistently provide context for these ideas, highlighting the
importance of data sharing and open science for phylogenetics, biology and science.

\section*{Availability and requirements}

\textbf{Project name:} Physcraper

\textbf{Project home page:} \href{https://physcraper.readthedocs.io/en/latest/index.html}{https://physcraper.readthedocs.io/en/latest/index.html}

\textbf{Operating System:} Linux, Mac, Windows

\textbf{Programming Language:} Python

\textbf{Other requirements:} Dependencies

\textbf{License:} GNU

\textbf{Any restrictions to use by non-academics:} As specified by the License

\section*{Abbreviations}

\textbf{OpenTree:} The Open Tree of Life project

\textbf{TNRS:} Taxonomic Name Resolution Service

\textbf{MRCA:} Most Recent Common Ancestor

\textbf{BLAST:} Basic Local Alignment Search Tool

\textbf{NCBI:} USA National Center for Biodiversity Information

\textbf{GBIF:} Global Biodiversity Information Facility

\textbf{API:} Application Programming Interface

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Declarations}

\begin{backmatter}

\section*{Ethics approval and consent to participate}
Not applicable.

\section*{Consent for publication}
Not applicable.

\section*{Availability of data and materials}

Code and datasets generated and analysed during the current study are available in
the repositories ``physcraper" containing the source code,
\href{https://github.com/McTavishLab/physcraper}{https://github.com/McTavishLab/physcraper};
 ``physcraperex" containing the examples,
\href{https://github.com/McTavishLab/physcraperex}{https://github.com/McTavishLab/physcraperex};
and, ``physcraper\_ms" containing this reproducible manuscript,
\href{https://github.com/McTavishLab/physcraper\_ms}{https://github.com/McTavishLab/physcraper\_ms}.

\section*{Competing interests}
The authors declare that they have no competing interests.

\section*{Funding}
This research was supported by the grant ``Sustaining the Open Tree of Life", NSF ABI
No. 1759838, and ABI No. 1759846.
Computer time was provided by the Multi-Environment Research Computer for
Exploration and Discovery (MERCED) cluster from the University of California,
Merced (UCM), supported by the NSF Grant No. ACI-1429783.

\section*{Authors' contributions}
LLSR wrote manuscript, alignment code, documentation, performed analyses and developed examples; MK wrote code for ncbidataparser module, filtering of sequences per OTU and using offline blast searches, wrote documentation and tests; EJM conceived study, wrote most of the code, documentation and tests.
All authors contributed to the manuscript and gave final approval for publication.


\section*{Acknowledgements}
We thank the members of the OpenTree development team and the ``short bar" Science
and Engineering Building 1, UCM, joint lab paper discussion group for valuable
comments on this manuscript. We also thank the valuable comments of David Posada, Rutger Vos,
and an anonymous reviewer that greatly improved an earlier version of this manuscript.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  \caption{The Physcraper framework consists of four general steps. The software is fully
  described on its documentation website at \href{https://physcraper.readthedocs.io}{https://physcraper.readthedocs.io},
  along with installation instructions, function usage descriptions, examples and tutorials.
}
\end{figure}

\begin{figure}[h!]
  \caption{A) Phylogeny updated with Physcraper using a starting phylogeny and an alignment from
  \cite{gottlieb2005molecular} (Gottlieb2005 data in text).
  Tips in original alignment and new tips added with Physcraper are depicted in
  black and red, respectively. Physcraper obtained sequences from the GenBank
  database via local BLAST of all sequences from the Gottlieb2005 internal transcribed
  spacer DNA region (ITS) original alignment,
  filtered them following criteria described in section ``DNA sequence search and
  filtering", aligned them to the original alignment using MUSCLE, and performed a
  phylogenetic reconstruction using RAxML, with 100 bootstraps. B) Results of conflict
  analysis performed using OpenTree's conflict tool \cite{redelings2017supertree}.
  The Physcraper updated Gottlieb2005 phylogeny in (A) was compared to an \textit{Ilex}
  OpenTree synthetic subtree v. 12.3 \cite{ott727571} constructed using taxonomy of the genus
  as backbone and resolving branches based on phylogenetic data from the original
  Gottlieb2005 phylogeny.
}
\end{figure}


\end{backmatter}
\end{document}
