\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[]{article}
\usepackage{lmodern}
\usepackage{setspace}
\setstretch{2}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdftitle={Physcraper: a python package for continual update of evolutionary estimates using the Open Tree of Life},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin = 1in]{geometry}
\usepackage{longtable,booktabs}
% Fix footnotes in tables (requires footnote package)
\IfFileExists{footnote.sty}{\usepackage{footnote}\makesavenoteenv{longtable}}{}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{color}
\usepackage{hyperref}
\usepackage{caption}
\usepackage{blindtext}
\usepackage{url}
\usepackage[left]{lineno}
\linenumbers

\title{Physcraper: a python package for continual update of evolutionary estimates using the Open Tree of Life}
\author{}
\date{\vspace{-2.5em}}

\begin{document}
\maketitle

\textbf{1. Luna L. Sanchez Reyes}

School of Natural Sciences, University of California, Merced

email: \href{mailto:sanchez.reyes.luna@gmail.com}{\nolinkurl{sanchez.reyes.luna@gmail.com}}

\textbf{2. Martha Kandziora}

School of Natural Sciences, University of California, Merced

email: \href{mailto:martha.kandziora@mailbox.org}{\nolinkurl{martha.kandziora@mailbox.org}}

\textbf{3. Emily Jane McTavish}

School of Natural Sciences, University of California, Merced

email: \href{mailto:ejmctavish@gmail.com}{\nolinkurl{ejmctavish@gmail.com}}

\textbf{Correspondence address}: Science and Engineering Building 1, University of California, Merced, 5200 N. Lake Rd, Merced CA 95343

\textbf{Correspondence email}: \href{mailto:sanchez.reyes.luna@gmail.com}{\nolinkurl{sanchez.reyes.luna@gmail.com}}, \href{mailto:ejmctavish@gmail.com}{\nolinkurl{ejmctavish@gmail.com}}

\textbf{Running title}: Continually updated gene trees with Physcraper

\textbf{Word count}: 3170

\textbf{Manuscript prepared for submission to Methods in Ecology and Evolution}

\textbf{Article type}: Application

\newpage

\hypertarget{abstract}{%
\section{Abstract}\label{abstract}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Phylogenies are a key part of research in all areas of biology. Tools that automatize
  some parts of the process of phylogenetic reconstruction (mainly character matrix construction)
  have been developed for the advantage of both specialists in the field of phylogenetics and nonspecialists.
  However, interpretation of results, comparison with previously available phylogenetic
  hypotheses, and choosing of one phylogeny for downstream analyses and discussion still impose difficulties
  to one that is not a specialist either on phylogenetic methods or on a particular group of study.
\item
  Physcraper is an open‚Äêsource, command-line Python program that automatizes the update of published
  phylogenies by making use of public DNA sequence data and taxonomic information,
  providing a framework for comparison of published phylogenies with their updated versions.
\item
  Physcraper can be used by the nonspecialist, as a tool to generate phylogenetic
  hypothesis based on already available expert phylogenetic knowledge.
  Phylogeneticists and group specialists will find it useful as a tool to facilitate comparison
  of alternative phylogenetic hypotheses (topologies).
  \textbf{\emph{Is physcraper intended for the nonspecialist?? We have two types of nonspecialists:
  the ones that do not know about phylogenetic methods and the ones that might know
  about phylogenetic methods but do not know much about a certain biological group.}}
\item
  Physcraper implements node by node/topology comparison of the the original and the updated
  trees using the conflict API of OToL, and summarizes differences.
\item
  We hope the physcraper workflow demonstrates the benefits of opening results in phylogenetics and encourages researchers to strive for better data sharing practices.
\item
  Physcraper can be used with any OS. Detailed instructions for installation and
  use are available at \url{https://github.com/McTavishLab/physcraper}.
\end{enumerate}

\textbf{Keywords}: cross-connectivity, gene tree, open science, open tree of life, phylogeny, public database, python, reproducibility, taxonomy, updated alignment

\newpage

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

From molecular data to alignments and phylogenies, public biological data resources such as the GenBank database (Benson \emph{et al.} \protect\hyperlink{ref-benson2000genbank}{2000}; Wheeler \emph{et al.} \protect\hyperlink{ref-wheeler2000database}{2000}), the TreeBASE repository (Piel \emph{et al.} \protect\hyperlink{ref-piel2009treebase}{2009}) and the Open Tree of Life curating system (McTavish \emph{et al.} \protect\hyperlink{ref-mctavish2015phylesystem}{2015}), are still accumulating data, and there is currently no straightforward way to automatically connect their data.

More than a decade ago, the National Center for Biodiversity Information (NCBI) molecular database, GenBank, released its version number 159 (April 15, 2007). With 72 million DNA sequences, it was estimated to have the potential to resolve evolutionary relationships of most of the 241 000 distinct taxa represented in it (about 98.05\% of taxa in the NCBI taxonomy; Sanderson \emph{et al.} \protect\hyperlink{ref-sanderson2008phylota}{2008}), which in turns coveres about 10\% of extant described biodiversity (taking a conservative estimate of extant diversity; Scott \protect\hyperlink{ref-scott2011ncbi}{2011}; Federhen \protect\hyperlink{ref-federhen2003taxonomy}{2003}).
In comparison, the current GenBank release number 238 (June 15, 2020) has tripled in size, hosting data for more than 217 million DNA sequences \href{https://ftp.ncbi.nih.gov/genbank/gbrel.txt}{(See GenBank's release website)}. Yet, publicly avilable phylogenetic relationships still cover about 90 000 taxa {[}Hinchliff \emph{et al.} (\protect\hyperlink{ref-hinchliff2015synthesis}{2015}); CURRENT SYNTH TREE CITATION{]}, covering less than one third of the taxonomic diversity with data avilable a decade ago. While it is true that many phylogenies are not publicly shared (Drew \emph{et al.} \protect\hyperlink{ref-drew2013lost}{2013}; Magee \emph{et al.} \protect\hyperlink{ref-magee2014dawn}{2014}), most recently published large trees have been made available in recent years, indicating a lag between the amount of new DNA data generated and its analysis in a phylogenetic context.

Many useful computational tools have been developed in an effort to make sense of the large amount of data in public molecular databases, as well as private ones. Generally refered to as ``pipelines'', most of these tools were motivated by the genomics revolution, to help researchers to deal with the massive amount of data, and automatically identify clusters of homologs for genomic assembly. Notably, this homolog DNA clusters can be used as homology hypotheses (i.e., molecular alignments) to reconstruct phylogenetic relationships.
Pipelines that automatize the assembly of DNA alignments from the GenBank database for phylogenetic reconstruction (``phylogenetic pipelines'') such as PHYLOTA (Sanderson \emph{et al.} \protect\hyperlink{ref-sanderson2008phylota}{2008}), PHLAWD (Smith \emph{et al.} \protect\hyperlink{ref-smith2009mega}{2009}), and SUPERSMART (Antonelli \emph{et al.} \protect\hyperlink{ref-antonelli2017toward}{2017}), have been widely applied to study the evolutionary relationships among different organisms (TABLE?? maybe supplementary data), from a phylogeny of no more than 20 species of the family of barracuda fish (Sphyraenidae; Santini \& Sorenson \protect\hyperlink{ref-santini2013first}{2013}), to a mega-phylogeny of almost 3 000 species of living ferns (Moniloformopses; Lehtonen \protect\hyperlink{ref-lehtonen2011towards}{2011}).

Pipelines have been an important incorporation to the field of phylogenetics in many ways, particularly because they represent a clear step towards reproducibility in the field. In contrast, most published phylogenies to date have been infered using alignments that have been assembled and curated ``by hand'' (Morrison \protect\hyperlink{ref-morrison2009would}{2009}).
There seems to be a preference in the field of the classic phylogenetics approach (few markers thoughtfully curated) over the genomics approach (a massive amount of DNA markers that will overcome potential errors in the alignment coming from a lack of human curation).
It has also been suggested that manual curation of classic alignments produces better phylogenetic reconstructions and this has been demonstrated for genomic alignments (Fragoso-Mart√≠nez \emph{et al.} \protect\hyperlink{ref-fragoso2017pilot}{2017}).

A way to incorporate the best of two worlds (massive amounts of newly released molecular data AND fine curation from human experts) would be to rely on published manually curated homology hypotheses as ``jump-start'' alignments (Morrison \protect\hyperlink{ref-morrison2006multiple}{2006}). This expert-curated alignments can be continuously enriched and updated by incorporating newly released data from public molecular databases.

As of April 2014, the TreeBASE repository hosted about 8 200 curated alignments, providing information on evolutionary relationships of almost 105 000 distinct taxa \href{https://www.treebase.org/treebase-web/home.html\#:~:text=TreeBASE\%20is\%20produced\%20and\%20governed,mapped\%20to\%20104\%2C593\%20distinct\%20taxa.}{(see TreeBASE's website about)}.
This database provides an untapped source of valuable expert knowledge with the potential to update phylogenetic relationships in several different regions of the tree of life.

The OToL tree repository (phylesystem; McTavish \emph{et al.} \protect\hyperlink{ref-mctavish2015phylesystem}{2015}) automatically incorporates phylogenies uploaded into TreeBASE, and stores metadata linking the tree to its corresponding alignment repository in TreeBASE. However, if there are multiple alignments, TreeBASE does not indicate how they were used to generate the tree. This provides a loose means of linking the tree with the exact alignment that generated it.

Ultimately, linkage of original alignment with its corresponding phylogeny has to be done by a human curator.
Moreover, different data repositories follow different systems for taxon and study identification, posing a real challenge to automatically link data from across databases that belong to the same taxon and study.
OToL's metadata system incorporates taxon identifiers from a variety of taxonomies and repositories, including the NCBI taxonomy, GBIF, etc., providing a way to automatically link data from different databases.

Physcraper is a python pipeline that relies on the OToL metadata system to connect databases through taxon identification numbers.
It's main functionality is to connect phylogenies stored in the OToL phylesystem, with alignments from TreeBASE and newly released DNA data from GenBank, in order to update previously known phylogenetic relationships in a continuous manner.
Because of its design, it allows taking advantage of the many resources provided by the OToL.
For example, it allows automatizing and standardizing the comparison of phylogenetic hypotheses with currently known relationships.

This is an effort to keep on directing ourselves towards a fully reproducible workflow in phylogenetics.
And an effort to more effectively make big data connections for the advantage of phylogenetics and biology in general.

\hypertarget{how-does-physcraper-work}{%
\section{How does Physcraper work?}\label{how-does-physcraper-work}}

\hypertarget{the-input-a-tree-and-an-alignment}{%
\subsection{The input: a tree and an alignment}\label{the-input-a-tree-and-an-alignment}}

\begin{itemize}
\tightlist
\item
  In order to take advantage of the OToL tools, the input tree needs to be stored in the
  OToL phylesystem \href{https://github.com/opentreeoflife/phylesystem}{github.com/opentreeoflife/phylesystem}.
  Currently, only trees connected to
  a published study can be stored in the phylesystem (although there are plans to
  allow storage of non published trees). The main
  reason for this is that trees in phylesystem have a set of user defined characteristics
  that are essential for automatizing the phylogeny update process. The most relevant of these being the standardization of taxonomic names and the definition of ingroup and outgroup. Outgroup and ingroup taxa in the original tree are identified and tagged. This allows to automatically set the root for the updated tree on the final steps of the pipeline.
  A user can choose from among the 1216 published trees supporting the resolved nodes of the synthetic tree in the OToL website (\href{https://tree.opentreeoflife.org/about/synthesis-release/v12.3}{See OToL's website about}). If the published tree you are interested in updating is not in there, you can upload it via OToL's curator tool (\href{https://tree.opentreeoflife.org/curator}{www.opentreeoflife.org/curator}.
\item
  The alignment should be a gene alignment that was used to generate the tree. The original
  alignments are usually stored in a public repository such as TreeBase (Piel \emph{et al.} \protect\hyperlink{ref-piel2009treebase}{2009}; Vos \emph{et al.} \protect\hyperlink{ref-vos2012nexml}{2012}),
  DRYAD (\href{http://datadryad.org/}{www.datadryad.org}), or the journal were the tree was originally published.
  If the alignment is stored in TreeBase, \texttt{physcraper} can download it directly,
  either from the TreeBASE website (\href{https://treebase.org/}{www.treebase.org})
  or through the TreeBASE GitHub repository (SuperTreeBASE; \href{https://github.com/TreeBASE/supertreebase}{github.com/TreeBASE/supertreebase}).
  If the alignment is on another repository, or constitutes personal data, a path to a local copy of the alignment has to be provided.
\item
  A taxon name matching step is performed to verify that all taxon names on the tips
  of the tree are in the DNA character matrix and vice versa.
\item
  A ``.csv'' file with the summary of taxon name matching is produced for the user.
\item
  Unmatched taxon names are dropped from both the tree and alignment.
  Technically, just one matching name is needed to perform the searches. Please, see next section.
\item
  A ``.tre'' file and a ``.fas'' file containing only the matched taxa are generated and saved in the \texttt{inputs} folder to be used in the following steps.
\end{itemize}

\hypertarget{dna-sequence-search-and-cleaning}{%
\subsection{DNA sequence search and cleaning}\label{dna-sequence-search-and-cleaning}}

\begin{itemize}
\item
  Technically, any DNA molecular database can be used to search for new sequences.
  However, we rely on the GenBank database.
  The new sequence search can be performed on the remote database or in a local database.
\item
  The next step is to identify the search taxon within the reference taxonomy. The search taxon will be used to constraint the DNA sequence search on the nucleotide database within that taxonomic group. Because we are using the NCBI nucleotide database, by default the reference taxonomy is the NCBI taxonomy. The search taxon can be determined by the user. If none is provided, then the search taxon is identified as the Most Recent Common Ancestor (MRCA) of the
  matched taxa belonging to the ingroup in the tree, that is also a named clade in the reference taxonomy. This is known as the Most Recent Common
  Ancestral Taxon (MRCAT; also referred in the literature as the Least Inclusive Common Ancestral Taxon - LICA).
  The MRCAT can be different from the phylogenetic MRCA when the latter is an unnamed clade in the reference taxonomy.
  To automatically identify the MRCAT of a group of taxon names, we make use of the OToL taxonomy tool (\url{https://github.com/OpenTreeOfLife/germinator/wiki/Taxonomy-API-v3\#mrca}). I FORGOT TO ASK YOU ABOUT THIS CITATION!

  Users can provide a search taxon that is either a more or a less inclusive
  clade relative to the ingroup of the original phylogeny. If the search taxon is more inclusive, the sequence search will be performed outside the MRCAT of the matched taxa, e.g., including all taxa within
  the family or the order that the ingroup belongs to. If the search taxon is a less inclusive clade, the users can focus on enriching a particular clade/region within the ingroup of the phylogeny.
\item
  The Basic Local Alignment Search Tool, BLAST (Altschul \emph{et al.} \protect\hyperlink{ref-altschul1990basic}{1990}, \protect\hyperlink{ref-altschul1997gapped}{1997}) is used to identify
  similarity between DNA sequences within the search taxon in a nucleotide
  database, and the sequences on the checked alignment.
  The \texttt{blastn} function from the BLAST command line tools (Camacho \emph{et al.} \protect\hyperlink{ref-camacho2009blast}{2009}) is used for local-database sequence searches.
  For remote-database searches, we modified the BioPython (Cock \emph{et al.} \protect\hyperlink{ref-cock2009biopython}{2009}) BLAST function in the \href{https://biopython.org/DIST/docs/api/Bio.Blast.NCBIWWW-module.html}{NCBIWWW module} to accept an alternate BLAST URL. This is useful when a user has no access to the computer capacity needed to setup a local database, and a local blast database can be set up on a remote machine to BLAST without NCBI's required wait times.
\item
  A pairwise BLAST search is performed. This means that each sequence
  in the alignment is BLASTed against DNA sequences in a nucleotide database constrained to the search
  taxon. Results from each one of these BLAST runs are recorded, and matched sequences are saved
  along with their corresponding identification numbers (accesion numbers in the case of the GenBank database). This information will be used later to store the whole sequences in a dedicated library within the physcraper folder, allowing for secondary analyses to run significantly faster.
\item
  Matched sequences will be discarded if the fall below a default e-value of 0.00001, and outside a default minimum and maximum length of 80\% and 120\%, respectively, of the average length of sequences in the checked alignment (gaps dropped).
  This filtering guarantees that genomic sequences are not included.
  All acepted sequences are assigned an internal identifier, and are further filtered.
\item
  Because the original alignments usually lack taxonomic identification numbers, an extra filtering
  step is needed. Accepted sequences that belong to the
  same taxon of the query sequence, and that are either identical or shorter than
  the original sequence are discarded. Only longer sequences belonging to the
  same taxon as the orignal sequence will be considered further for analysis.
\item
  Among the remaining filtered sequences, there are usually several exemplars per taxon.
  Although it can be useful to keep some of them, for example, to investigate monophyly
  within species, there can be hundreds of exemplar sequences per taxon for some markers.
  To control the number of sequences per taxon in downstream analyses,
  5 sequences per taxon are chosen at random. This number is set by default but can be modified by the user.
\item
  Reverse complement sequences are identified and translated.
\item
  Users can choose to perform a more ``cycles'' of sequence similarity search, by blasting the newly found sequences. This can be done iteratively, but by default only sequences in the checked alignment are blasted. I ALSO FORGOT TO ASK: \textbf{\emph{Is there an argument to control the number of cycles of blast searches with new sequences?}}
\item
  Accepted sequences are downloaded in full, and stored as a local database in a directory that is globally accesible (default to physcraper/taxonomy), so they are accesible for further runs.
\item
  A fasta file containing all filtered and processed sequences resulting from the BLAST search is generated for the user.
\end{itemize}

\hypertarget{dna-sequence-alignment}{%
\subsection{DNA sequence alignment}\label{dna-sequence-alignment}}

\begin{itemize}
\tightlist
\item
  The software MUSCLE (Edgar \protect\hyperlink{ref-edgar2004muscle}{2004}) is implemented to perform sequence alignments.
\item
  First, all new sequences are aligned using default MUSCLE options.
\item
  Then, a MUSCLE profile alignment is performed, in which the original alignment
  is used as a template to align new sequences. This ensures that the final alignment
  follows the homology criteria established by the original alignment.
\item
  The final alignment is not further processed automatically. So, we encourage users
  to check it either by eye and perform manual refinement or using any of the many
  tools for alignment processing, to eliminate columns with no information.
\end{itemize}

\hypertarget{tree-reconstruction-and-comparison}{%
\subsection{Tree reconstruction and comparison}\label{tree-reconstruction-and-comparison}}

\begin{itemize}
\tightlist
\item
  A gene tree is reconstructed for each alignment provided, using a Maximum Likelihood approach implemented with the software RAxML (Stamatakis \protect\hyperlink{ref-stamatakis2014raxml}{2014}), using default settings such as a GTRCAT model of molecular evolution and 100 bootstrap replicates with default method. Currently only the number of bootsrap replicates can be modified by the user.
\item
  The original tree is used as starting tree for the ML searches. It can also be set as a full topological constraint or not used at all, depending on the goals of the user.
\item
  Bootstrap results are summarized with Dendropy (Sukumaran \& Holder \protect\hyperlink{ref-sukumaran2010dendropy}{2010})
\item
  The final result is an updated phylogenetic hypothesis for each of the genes provided in the alignment.
\item
  Tips on all trees generated by physcraper are defined by a taxon name space, allowing to perform comparisons and conflict analyses.
\item
  Robinson Foulds weighted and unweighted metrics are calculated with Dendropy functions.
\item
  Finally a conflict analysis is performed. This is basically a node by node comparison between the the synthetic OToL and the original and updated tree individually. This is performed with OToL's conflict Application Programming Interface (Redelings \& Holder \protect\hyperlink{ref-redelings2017supertree}{2017}).
\item
  For the conflict analysis to be meaningful, the root of the tree ineeds to be accurately defined.
\item
  A suggested default rooting based on OToL's taxonomy is implemented for now. It takes one descendant from the corresponding induced subtree in OToL's synthetic tree and then finds the MRCA of those in th eupdated trees. However, if the updated tree changes expectations from taxonomy, the root will no longer be accurate. Automatic identification of a phylogenetic tree root is indeed a difficult problem that has not been solved yet. The best way right now is for users to define outgroups so trees are accurately rooted.
\end{itemize}

\hypertarget{examples}{%
\section{Examples}\label{examples}}

We will illustrate the utility of physcraper in here with two use-case scenarios. One in which the user is interested in a particular group. Another one in which the user is interested in a particular phylogeny.
A tutorial as well as illustrated examples of commands for every step needed to perform a physcraper analysis are available elsewhere.

\hypertarget{the-hollies}{%
\subsection{The hollies}\label{the-hollies}}

A student is interested in the genus \emph{Ilex}, the only extant clade within the family Aquifoliaceae, order Aquifoliales of flowering plants.
The genus encompasses between 400-600 living species. A review of literature reveals that there are three published phylogenetic trees showing relationships within the hollies.
The first one has been made available in TreeBASE as well as in the OToL phylesystem and is part of the synthetic tree. It samples 48 species.
The second tree has not been made available anywhere, not even in the supplementary data of the original publication.
The most recent one has been made available in the OToL Phylesystem and in the DRYAD repository. It is the best sampled yet, with 200 species. However, it has not been added to the syntehtic tree yet.
This also makes it a perfect case to test the basic functionalities of physcraper: we know that the sequences of the most recently published tree have been made available on the GenBank database. Hence, we expect that updating the oldest tree will produce something very similar to the newest tree.

DESCRIBE RESULTS: SUMMARY OF NEW TAXA FOUND RELATIVE TO ORIGINAL TREE AND RELATIVE TO OTOL
RF DISTANCE INTERPRETATION
HOW MUCH TIME THE BLAST RUN TOOK
ML ESTIMATES OF UPDATED TREE VS ORIGINAL TREE

FIGURE: FACE TO FACE ORIGINAL VS UPDATED PHYLOGENY, IN RED NEW TAXA NOT IN OTOL.

\hypertarget{the-malvaceae}{%
\subsection{The Malvaceae}\label{the-malvaceae}}

A postdoc started working with a new reserach group. They are interested in solving relationships among lineages of the Malvaceae, a family of flowering plants with almost 6 000 known species, containing the relatives of cacao, cotton, durian and okra.
A review of the literature shows them that there are many phylogenetic trees encompassing some of the linegaes in the group. However, the head of the research group wants to use a particular marker they believe to be the best one to be able to solve the relationships in the group. They have been working in the alignment for long and they want to incorporate new data into the hypothesis of homology that they have been curating and that they trust.

\begin{figure}

{\centering \includegraphics[width=0.85\linewidth]{docs/figs/cotree-plot2-1} 

}

\caption{Comparison of original tree and update tree of the Malvaceae.}\label{fig:fig-2}
\end{figure}
\newpage

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

Data repositories hold more information than meets the eye.
Besides the actual data, they are rich sources of information that can be used for the advantage of biology and science in general.

Many pipelines are making use of DNA data repositories in different ways.
Most of them focus on efficient ways to mine the data -- getting the most homologs.
Some focus on accurate ways of mining the data -- getting real and clean homologs.
Others focus on refining the alignment.
Most focus on generating full trees \emph{de novo}, mainly for regions of the Tree of
Life that have no phylogenetic assessment yet in published studies, but also for
regions that have been already studied and that have phylogenetic data already.
However, expert phylogenetic knowledge is also an important source of data in public
and open repositories that is potentiallly being diregarded.

All these tools are key efforts for advancing towards reproducibility in phylogenetics,
a field that has been largely recognised as somewhat artisanal.
In here, we highlight the potential of other sources of information available from data repositories and present a method to link data from different repositories, while leveraging on the knowledge and intuition of the expert
community to build up our phylogenetic knowledge, using not only data accumulated in
molecular data repositories, but phylogenetic knowledge accumulated in phylogenetic tree repositories.
While not generating full phylogenies \emph{de novo}, pyscraper is still capable of generating new phylogenetic knowledge.
In this way, the physcraper workflow can be complemented with other pipelines, to potentially increase speed (IS THIS TRUE??).
It can also be combined with data from repositories other than molecular data. For example geographic locations (GBIF), fossils (PBDB), etc.

Physcraper has the potential to be applied for the advantage of the field to rapidly
place newly discovered species phylogenetically (Webb \emph{et al.} \protect\hyperlink{ref-webb2010biodiversity}{2010}),
obtain trees for ecophylogenetic studies (Helmus \& Ives \protect\hyperlink{ref-helmus2012phylogenetic}{2012}),
help to sistematize molecular databases, i.e., curate taxonomic assignations (San Mauro \& Agorreta \protect\hyperlink{ref-san2010molecular}{2010}),
and rapidly generate custom species trees for downstream analyses (Stoltzfus \emph{et al.} \protect\hyperlink{ref-stoltzfus2013phylotastic}{2013}).

\hypertarget{acknowledgements}{%
\section{Acknowledgements}\label{acknowledgements}}

The University of California, Merced cluster, MERCED (Multi-Environment Research Computer for Exploration and Discovery) supported by the National Science Foundation (Grant No.~ACI-1429783).

We acknowledge contributions from \ldots{}

\hypertarget{authors-contributions}{%
\section{Authors' Contributions}\label{authors-contributions}}

\hypertarget{data-avilability}{%
\section{Data Avilability}\label{data-avilability}}

\hypertarget{references}{%
\section{References}\label{references}}

\newpage
\begin{center}
\textsc{References}
\end{center}

\hypertarget{refs}{}
\leavevmode\hypertarget{ref-altschul1990basic}{}%
Altschul, S.F., Gish, W., Miller, W., Myers, E.W. \& Lipman, D.J. (1990). Basic local alignment search tool. \emph{Journal of molecular biology}, \textbf{215}, 403--410.

\leavevmode\hypertarget{ref-altschul1997gapped}{}%
Altschul, S.F., Madden, T.L., Sch√§ffer, A.A., Zhang, J., Zhang, Z., Miller, W. \& Lipman, D.J. (1997). Gapped blast and psi-blast: A new generation of protein database search programs. \emph{Nucleic acids research}, \textbf{25}, 3389--3402.

\leavevmode\hypertarget{ref-antonelli2017toward}{}%
Antonelli, A., Hettling, H., Condamine, F.L., Vos, K., Nilsson, R.H., Sanderson, M.J., Sauquet, H., Scharn, R., Silvestro, D., T√∂pel, M. \& others. (2017). Toward a self-updating platform for estimating rates of speciation and migration, ages, and relationships of taxa. \emph{Systematic Biology}, \textbf{66}, 152--166.

\leavevmode\hypertarget{ref-benson2000genbank}{}%
Benson, D.A., Karsch-Mizrachi, I., Lipman, D.J., Ostell, J., Rapp, B.A. \& Wheeler, D.L. (2000). GenBank. \emph{Nucleic acids research}, \textbf{28}, 15--18.

\leavevmode\hypertarget{ref-camacho2009blast}{}%
Camacho, C., George, C., Vahram, A., Ning, M., Jason, P., Kevin, B. \& Thomas, L. (2009). BLAST+: Architecture and applications. \emph{BMC bioinformatics}, \textbf{10}, 421.

\leavevmode\hypertarget{ref-cock2009biopython}{}%
Cock, P.J., Antao, T., Chang, J.T., Chapman, B.A., Cox, C.J., Dalke, A., Friedberg, I., Hamelryck, T., Kauff, F., Wilczynski, B. \& others. (2009). Biopython: Freely available python tools for computational molecular biology and bioinformatics. \emph{Bioinformatics}, \textbf{25}, 1422--1423.

\leavevmode\hypertarget{ref-drew2013lost}{}%
Drew, B.T., Gazis, R., Cabezas, P., Swithers, K.S., Deng, J., Rodriguez, R., Katz, L.A., Crandall, K.A., Hibbett, D.S. \& Soltis, D.E. (2013). Lost branches on the tree of life. \emph{PLoS biology}, \textbf{11}.

\leavevmode\hypertarget{ref-edgar2004muscle}{}%
Edgar, R.C. (2004). MUSCLE: Multiple sequence alignment with high accuracy and high throughput. \emph{Nucleic acids research}, \textbf{32}, 1792--1797.

\leavevmode\hypertarget{ref-federhen2003taxonomy}{}%
Federhen, S. (2003). The taxonomy project. \emph{The NCBI Handbook}.

\leavevmode\hypertarget{ref-fragoso2017pilot}{}%
Fragoso-Mart√≠nez, I., Salazar, G.A., Mart√≠nez-Gordillo, M., Magall√≥n, S., S√°nchez-Reyes, L., Lemmon, E.M., Lemmon, A.R., Sazatornil, F. \& Mendoza, C.G. (2017). A pilot study applying the plant anchored hybrid enrichment method to new world sages (salvia subgenus calosphace; lamiaceae). \emph{Molecular Phylogenetics and Evolution}, \textbf{117}, 124--134.

\leavevmode\hypertarget{ref-helmus2012phylogenetic}{}%
Helmus, M.R. \& Ives, A.R. (2012). Phylogenetic diversity--area curves. \emph{Ecology}, \textbf{93}, S31--S43.

\leavevmode\hypertarget{ref-hinchliff2015synthesis}{}%
Hinchliff, C.E., Smith, S.A., Allman, J.F., Burleigh, J.G., Chaudhary, R., Coghill, L.M., Crandall, K.A., Deng, J., Drew, B.T., Gazis, R. \& others. (2015). Synthesis of phylogeny and taxonomy into a comprehensive tree of life. \emph{Proceedings of the National Academy of Sciences}, \textbf{112}, 12764--12769.

\leavevmode\hypertarget{ref-lehtonen2011towards}{}%
Lehtonen, S. (2011). Towards resolving the complete fern tree of life. \emph{PLoS One}, \textbf{6}.

\leavevmode\hypertarget{ref-magee2014dawn}{}%
Magee, A.F., May, M.R. \& Moore, B.R. (2014). The dawn of open access to phylogenetic data. \emph{PLoS One}, \textbf{9}.

\leavevmode\hypertarget{ref-mctavish2015phylesystem}{}%
McTavish, E.J., Hinchliff, C.E., Allman, J.F., Brown, J.W., Cranston, K.A., Holder, M.T., Rees, J.A. \& Smith, S.A. (2015). Phylesystem: A git-based data store for community-curated phylogenetic estimates. \emph{Bioinformatics}, \textbf{31}, 2794--2800.

\leavevmode\hypertarget{ref-morrison2006multiple}{}%
Morrison, D.A. (2006). Multiple sequence alignment for phylogenetic purposes. \emph{Australian Systematic Botany}, \textbf{19}, 479--539.

\leavevmode\hypertarget{ref-morrison2009would}{}%
Morrison, D.A. (2009). Why would phylogeneticists ignore computerized sequence alignment? \emph{Systematic biology}, \textbf{58}, 150--158.

\leavevmode\hypertarget{ref-piel2009treebase}{}%
Piel, W., Chan, L., Dominus, M., Ruan, J., Vos, R. \& Tannen, V. (2009). Treebase v. 2: A database of phylogenetic knowledge. E-biosphere.

\leavevmode\hypertarget{ref-redelings2017supertree}{}%
Redelings, B.D. \& Holder, M.T. (2017). A supertree pipeline for summarizing phylogenetic and taxonomic information for millions of species. \emph{PeerJ}, \textbf{5}, e3058.

\leavevmode\hypertarget{ref-sanderson2008phylota}{}%
Sanderson, M.J., Boss, D., Chen, D., Cranston, K.A. \& Wehe, A. (2008). The PhyLoTA Browser: Processing GenBank for Molecular Phylogenetics Research. \emph{Systematic Biology}, \textbf{57}, 335--346.

\leavevmode\hypertarget{ref-san2010molecular}{}%
San Mauro, D. \& Agorreta, A. (2010). Molecular systematics: A synthesis of the common methods and the state of knowledge. \emph{Cellular \& Molecular Biology Letters}, \textbf{15}, 311.

\leavevmode\hypertarget{ref-santini2013first}{}%
Santini, F. \& Sorenson, L. (2013). First molecular timetree of billfishes (istiophoriformes: Acanthomorpha) shows a late miocene radiation of marlins and allies. \emph{Italian journal of zoology}, \textbf{80}, 481--489.

\leavevmode\hypertarget{ref-scott2011ncbi}{}%
Scott, F. (2011). The ncbi taxonomy database. \emph{Nucleic Acids Research}, \textbf{40}, D136--D14.

\leavevmode\hypertarget{ref-smith2009mega}{}%
Smith, S.A., Beaulieu, J.M. \& Donoghue, M.J. (2009). Mega-phylogeny approach for comparative biology: An alternative to supertree and supermatrix approaches. \emph{BMC evolutionary biology}, \textbf{9}, 37.

\leavevmode\hypertarget{ref-stamatakis2014raxml}{}%
Stamatakis, A. (2014). RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. \emph{Bioinformatics}, \textbf{30}, 1312--1313.

\leavevmode\hypertarget{ref-stoltzfus2013phylotastic}{}%
Stoltzfus, A., Lapp, H., Matasci, N., Deus, H., Sidlauskas, B., Zmasek, C.M., Vaidya, G., Pontelli, E., Cranston, K., Vos, R. \& others. (2013). Phylotastic! Making tree-of-life knowledge accessible, reusable and convenient. \emph{BMC bioinformatics}, \textbf{14}, 158.

\leavevmode\hypertarget{ref-sukumaran2010dendropy}{}%
Sukumaran, J. \& Holder, M.T. (2010). DendroPy: A python library for phylogenetic computing. \emph{Bioinformatics}, \textbf{26}, 1569--1571.

\leavevmode\hypertarget{ref-vos2012nexml}{}%
Vos, R.A., Balhoff, J.P., Caravas, J.A., Holder, M.T., Lapp, H., Maddison, W.P., Midford, P.E., Priyam, A., Sukumaran, J., Xia, X. \& others. (2012). NeXML: Rich, extensible, and verifiable representation of comparative data and metadata. \emph{Systematic biology}, \textbf{61}, 675--689.

\leavevmode\hypertarget{ref-webb2010biodiversity}{}%
Webb, C.O., Slik, J.F. \& Triono, T. (2010). Biodiversity inventory and informatics in southeast asia. \emph{Biodiversity and Conservation}, \textbf{19}, 955--972.

\leavevmode\hypertarget{ref-wheeler2000database}{}%
Wheeler, D.L., Chappey, C., Lash, A.E., Leipe, D.D., Madden, T.L., Schuler, G.D., Tatusova, T.A. \& Rapp, B.A. (2000). Database resources of the national center for biotechnology information. \emph{Nucleic acids research}, \textbf{28}, 10--14.

\end{document}
